diff -Nur linux-4.19.75.orig/drivers/net/ethernet/freescale/fec_main.c linux-4.19.75/drivers/net/ethernet/freescale/fec_main.c
--- linux-4.19.75.orig/drivers/net/ethernet/freescale/fec_main.c	2019-09-21 07:17:15.000000000 +0200
+++ linux-4.19.75/drivers/net/ethernet/freescale/fec_main.c	2019-10-10 01:17:45.284188283 +0200
@@ -156,7 +156,7 @@
 module_param_array(macaddr, byte, NULL, 0);
 MODULE_PARM_DESC(macaddr, "FEC Ethernet MAC address");
 
-#if defined(CONFIG_M5272)
+#if defined(CONFIG_COLDFIRE)
 /*
  * Some hardware gets it MAC address out of local flash memory.
  * if this is non-zero then assume it is the address to get MAC from.
@@ -174,7 +174,7 @@
 #else
 #define	FEC_FLASHMAC	0
 #endif
-#endif /* CONFIG_M5272 */
+#endif /* CONFIG_COLDFIRE */
 
 /* The FEC stores dest/src/type/vlan, data, and checksum for receive packets.
  *
@@ -958,7 +958,7 @@
 	/* Set MII speed */
 	writel(fep->phy_speed, fep->hwp + FEC_MII_SPEED);
 
-#if !defined(CONFIG_M5272)
+#if !defined(CONFIG_COLDFIRE)
 	if (fep->quirks & FEC_QUIRK_HAS_RACC) {
 		val = readl(fep->hwp + FEC_RACC);
 		/* align IP header */
@@ -1027,7 +1027,7 @@
 #endif
 	}
 
-#if !defined(CONFIG_M5272)
+#if !defined(CONFIG_COLDFIRE)
 	/* enable pause frame*/
 	if ((fep->pause_flag & FEC_PAUSE_FLAG_ENABLE) ||
 	    ((fep->pause_flag & FEC_PAUSE_FLAG_AUTONEG) &&
@@ -1045,13 +1045,13 @@
 	} else {
 		rcntl &= ~FEC_ENET_FCE;
 	}
-#endif /* !defined(CONFIG_M5272) */
+#endif /* !defined(CONFIG_COLDFIRE) */
 
 	writel(rcntl, fep->hwp + FEC_R_CNTRL);
 
 	/* Setup multicast filter. */
 	set_multicast_list(ndev);
-#ifndef CONFIG_M5272
+#ifndef CONFIG_COLDFIRE
 	writel(0, fep->hwp + FEC_HASH_TABLE_HIGH);
 	writel(0, fep->hwp + FEC_HASH_TABLE_LOW);
 #endif
@@ -1066,7 +1066,7 @@
 	if (fep->bufdesc_ex)
 		ecntl |= (1 << 4);
 
-#ifndef CONFIG_M5272
+#ifndef CONFIG_COLDFIRE
 	/* Enable the MIB statistic event counters */
 	writel(0 << 31, fep->hwp + FEC_MIB_CTRLSTAT);
 #endif
@@ -1664,7 +1664,7 @@
 	 * 3) from flash or fuse (via platform data)
 	 */
 	if (!is_valid_ether_addr(iap)) {
-#ifdef CONFIG_M5272
+#ifdef CONFIG_COLDFIRE
 		if (FEC_FLASHMAC)
 			iap = (unsigned char *)FEC_FLASHMAC;
 #else
@@ -1943,7 +1943,7 @@
 	if (fep->quirks & FEC_QUIRK_HAS_GBIT) {
 		phy_dev->supported &= PHY_GBIT_FEATURES;
 		phy_dev->supported &= ~SUPPORTED_1000baseT_Half;
-#if !defined(CONFIG_M5272)
+#if !defined(CONFIG_COLDFIRE)
 		phy_dev->supported |= SUPPORTED_Pause;
 #endif
 	}
@@ -2199,7 +2199,7 @@
 	}
 }
 
-#if !defined(CONFIG_M5272)
+#if !defined(CONFIG_COLDFIRE)
 
 static void fec_enet_get_pauseparam(struct net_device *ndev,
 				    struct ethtool_pauseparam *pause)
@@ -2391,7 +2391,7 @@
 static inline void fec_enet_clear_ethtool_stats(struct net_device *dev)
 {
 }
-#endif /* !defined(CONFIG_M5272) */
+#endif /* !defined(CONFIG_COLDFIRE) */
 
 /* ITR clock source is enet system clock (clk_ahb).
  * TCTT unit is cycle_ns * 64 cycle
@@ -2596,7 +2596,7 @@
 	.get_link		= ethtool_op_get_link,
 	.get_coalesce		= fec_enet_get_coalesce,
 	.set_coalesce		= fec_enet_set_coalesce,
-#ifndef CONFIG_M5272
+#ifndef CONFIG_COLDFIRE
 	.get_pauseparam		= fec_enet_get_pauseparam,
 	.set_pauseparam		= fec_enet_set_pauseparam,
 	.get_strings		= fec_enet_get_strings,
@@ -3385,7 +3385,7 @@
 	fep->num_rx_queues = num_rx_qs;
 	fep->num_tx_queues = num_tx_qs;
 
-#if !defined(CONFIG_M5272)
+#if !defined(CONFIG_COLDFIRE)
 	/* default enable pause frame auto negotiation */
 	if (fep->quirks & FEC_QUIRK_HAS_GBIT)
 		fep->pause_flag |= FEC_PAUSE_FLAG_AUTONEG;
