diff -Nur linux-4.17.orig/init/initramfs.c linux-4.17/init/initramfs.c
--- linux-4.17.orig/init/initramfs.c	2018-06-03 23:15:21.000000000 +0200
+++ linux-4.17/init/initramfs.c	2018-06-10 17:01:16.000000000 +0200
@@ -659,6 +659,9 @@
 	 * us a chance to load before device_initcalls.
 	 */
 	load_default_modules();
+#ifdef CONFIG_DEVTMPFS_MOUNT
+	devtmpfs_mount("dev");
+#endif
 
 	return 0;
 }
diff -Nur linux-4.17.orig/init/main.c linux-4.17/init/main.c
--- linux-4.17.orig/init/main.c	2018-06-03 23:15:21.000000000 +0200
+++ linux-4.17/init/main.c	2018-06-10 17:01:52.000000000 +0200
@@ -1138,6 +1138,8 @@
 	if (ksys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
 		pr_err("Warning: unable to open an initial console.\n");
 
+	printk(KERN_WARNING "Starting Linux (built with OpenADK).\n");
+
 	(void) ksys_dup(0);
 	(void) ksys_dup(0);
 	/*
@@ -1146,7 +1148,7 @@
 	 */
 
 	if (!ramdisk_execute_command)
-		ramdisk_execute_command = "/init";
+		ramdisk_execute_command = "/sbin/init";
 
 	if (ksys_access((const char __user *)
 			ramdisk_execute_command, 0) != 0) {
