diff -Nur linux-3.13.3.orig/init/main.c linux-3.13.3/init/main.c
--- linux-3.13.3.orig/init/main.c	2014-02-13 23:00:14.000000000 +0100
+++ linux-3.13.3/init/main.c	2014-02-17 11:35:14.000000000 +0100
@@ -916,6 +917,8 @@
 	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
 		pr_err("Warning: unable to open an initial console.\n");
 
+	printk(KERN_WARNING "Starting Linux (built with OpenADK).\n");
+
 	(void) sys_dup(0);
 	(void) sys_dup(0);
 	/*
diff -Nur linux-3.13.6.orig/init/initramfs.c linux-3.13.6/init/initramfs.c
--- linux-3.13.6.orig/init/initramfs.c	2014-03-07 07:07:02.000000000 +0100
+++ linux-3.13.6/init/initramfs.c	2014-03-15 12:11:31.882731916 +0100
@@ -622,6 +622,9 @@
 		 */
 		load_default_modules();
 	}
+#ifdef CONFIG_DEVTMPFS_MOUNT
+	devtmpfs_mount("dev");
+#endif
 	return 0;
 }
 rootfs_initcall(populate_rootfs);
diff -Nur linux-3.13.6.orig/init/main.c linux-3.13.6/init/main.c
--- linux-3.13.6.orig/init/main.c	2014-03-07 07:07:02.000000000 +0100
+++ linux-3.13.6/init/main.c	2014-03-15 12:13:16.459024452 +0100
@@ -924,7 +924,7 @@
 	 */
 
 	if (!ramdisk_execute_command)
-		ramdisk_execute_command = "/init";
+		ramdisk_execute_command = "/sbin/init";
 
 	if (sys_access((const char __user *) ramdisk_execute_command, 0) != 0) {
 		ramdisk_execute_command = NULL;
