diff -Nur linux-3.12.33.orig/drivers/base/regmap/Kconfig linux-3.12.33/drivers/base/regmap/Kconfig
--- linux-3.12.33.orig/drivers/base/regmap/Kconfig	2014-11-15 13:28:07.000000000 +0100
+++ linux-3.12.33/drivers/base/regmap/Kconfig	2014-12-04 11:18:59.146540398 +0100
@@ -10,13 +10,13 @@
 	bool
 
 config REGMAP_I2C
-	tristate
+	bool
 
 config REGMAP_SPI
-	tristate
+	bool
 
 config REGMAP_MMIO
-	tristate
+	bool
 
 config REGMAP_IRQ
 	bool
