diff -Nur linux-2.6.36.orig/arch/cris/include/arch-v10/arch/Kbuild linux-2.6.36/arch/cris/include/arch-v10/arch/Kbuild
--- linux-2.6.36.orig/arch/cris/include/arch-v10/arch/Kbuild	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36/arch/cris/include/arch-v10/arch/Kbuild	2010-11-15 17:35:24.000000000 +0100
@@ -1,4 +1,9 @@
+header-y += dma.h
+header-y += io_interface_mux.h
 header-y += user.h
 header-y += svinto.h
 header-y += sv_addr_ag.h
 header-y += sv_addr.agh
+header-y += elf.h
+header-y += page.h
+header-y += ptrace.h
diff -Nur linux-2.6.36.orig/arch/cris/include/arch-v32/arch/Kbuild linux-2.6.36/arch/cris/include/arch-v32/arch/Kbuild
--- linux-2.6.36.orig/arch/cris/include/arch-v32/arch/Kbuild	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36/arch/cris/include/arch-v32/arch/Kbuild	2010-11-15 17:35:24.000000000 +0100
@@ -1,2 +1,6 @@
 header-y += user.h
 header-y += cryptocop.h
+header-y += elf.h
+header-y += page.h
+header-y += ptrace.h
+
diff -Nur linux-2.6.36.orig/arch/cris/include/asm/Kbuild linux-2.6.36/arch/cris/include/asm/Kbuild
--- linux-2.6.36.orig/arch/cris/include/asm/Kbuild	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36/arch/cris/include/asm/Kbuild	2010-11-15 17:37:57.000000000 +0100
@@ -1,10 +1,13 @@
 include include/asm-generic/Kbuild.asm
 
-header-y += arch-v10/
-header-y += arch-v32/
-
+header-y += ../arch-v10/arch/
+header-y += ../arch-v32/arch/
+  
+header-y += elf.h
 header-y += ethernet.h
 header-y += etraxgpio.h
 header-y += rs485.h
 header-y += rtc.h
 header-y += sync_serial.h
+header-y += page.h
+header-y += user.h
diff -Nur linux-2.6.36.orig/include/asm-generic/Kbuild linux-2.6.36/include/asm-generic/Kbuild
--- linux-2.6.36.orig/include/asm-generic/Kbuild	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36/include/asm-generic/Kbuild	2010-11-15 17:37:02.000000000 +0100
@@ -3,14 +3,17 @@
 header-y += errno-base.h
 header-y += errno.h
 header-y += fcntl.h
+header-y += getorder.h
 header-y += int-l64.h
 header-y += int-ll64.h
 header-y += ioctl.h
 header-y += ioctls.h
 header-y += ipcbuf.h
+header-y += memory_model.h
 header-y += mman-common.h
 header-y += mman.h
 header-y += msgbuf.h
+header-y += page.h
 header-y += param.h
 header-y += poll.h
 header-y += posix_types.h
diff -Nur linux-2.6.36.orig/include/linux/Kbuild linux-2.6.36/include/linux/Kbuild
--- linux-2.6.36.orig/include/linux/Kbuild	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36/include/linux/Kbuild	2010-11-15 17:36:11.000000000 +0100
@@ -365,6 +365,7 @@
 header-y += un.h
 header-y += unistd.h
 header-y += usbdevice_fs.h
+header-y += user.h
 header-y += utime.h
 header-y += utsname.h
 header-y += veth.h
