diff -Nur uClibc-0.9.32-rc3.orig/Makefile.in uClibc-0.9.32-rc3/Makefile.in
--- uClibc-0.9.32-rc3.orig/Makefile.in	2011-03-16 20:11:13.000000000 +0100
+++ uClibc-0.9.32-rc3/Makefile.in	2011-03-18 18:00:13.000000000 +0100
@@ -383,6 +383,8 @@
 # Installs run-time libraries
 install_runtime: all | $(PREFIX)$(RUNTIME_PREFIX)$(MULTILIB_DIR)
 ifeq ($(HAVE_SHARED),y)
+	$(INSTALL) -m 755 $(top_builddir)lib/ld-uClibc-$(VERSION).so \
+		$(PREFIX)$(RUNTIME_PREFIX)$(MULTILIB_DIR)
 	$(INSTALL) -m 755 $(top_builddir)lib/lib*-$(VERSION).so \
 		$(PREFIX)$(RUNTIME_PREFIX)$(MULTILIB_DIR)
 	(cd $(top_builddir)lib && $(TAR) -cf - *.so.*) | $(TAR) -xf - -C $(PREFIX)$(RUNTIME_PREFIX)$(MULTILIB_DIR)
