diff -Nur uClibc-0.9.32-rc3.orig/Rules.mak uClibc-0.9.32-rc3/Rules.mak
--- uClibc-0.9.32-rc3.orig/Rules.mak	2011-03-16 20:11:13.000000000 +0100
+++ uClibc-0.9.32-rc3/Rules.mak	2011-05-08 21:55:43.986610132 +0200
@@ -120,7 +120,7 @@
 SHARED_LIBNAME := $(LIBC).so.$(ABI_VERSION)
 UBACKTRACE_DSO := libubacktrace.so.$(MAJOR_VERSION)
 ifneq ($(findstring  $(TARGET_ARCH) , hppa64 ia64 mips64 powerpc64 s390x sparc64 x86_64 ),)
-UCLIBC_LDSO_NAME := ld64-uClibc
+UCLIBC_LDSO_NAME := ld-uClibc
 ARCH_NATIVE_BIT := 64
 else
 UCLIBC_LDSO_NAME := ld-uClibc
