diff -Nur uClibc-0.9.32-rc1.orig/libc/inet/netlinkaccess.h uClibc-0.9.32-rc1/libc/inet/netlinkaccess.h
--- uClibc-0.9.32-rc1.orig/libc/inet/netlinkaccess.h	2010-12-17 20:05:17.000000000 +0100
+++ uClibc-0.9.32-rc1/libc/inet/netlinkaccess.h	2011-01-01 15:49:58.906931975 +0100
@@ -22,15 +22,8 @@
 #include <features.h>
 #include <stdint.h>
 #include <unistd.h>
-#include <sys/types.h>
-
 #if defined __ASSUME_NETLINK_SUPPORT || defined __UCLIBC_USE_NETLINK__
-#define _LINUX_TYPES_H
-typedef uint8_t __u8;
-typedef uint16_t __u16;
-typedef uint32_t __u32;
-typedef uint64_t __u64;
-typedef int32_t __s32;
+#include <asm/types.h>
 #include <linux/rtnetlink.h>
 #include <linux/netlink.h>
 
