diff -Nur uClibc-0.9.30.2.orig/libc/sysdeps/linux/sparc/sigaction.c uClibc-0.9.30.2/libc/sysdeps/linux/sparc/sigaction.c
--- uClibc-0.9.30.2.orig/libc/sysdeps/linux/sparc/sigaction.c	2010-01-13 22:22:31.000000000 +0100
+++ uClibc-0.9.30.2/libc/sysdeps/linux/sparc/sigaction.c	2010-02-14 12:10:52.000000000 +0100
@@ -34,7 +34,8 @@
 static void __rt_sigreturn_stub(void);
 static void __sigreturn_stub(void);
 
-int __libc_sigaction(int sig, const struct sigaction *act, struct sigaction *oact)
+libc_hidden_proto(sigaction)
+int sigaction(int sig, const struct sigaction *act, struct sigaction *oact)
 {
 	int ret;
 	struct sigaction kact, koact;
@@ -65,10 +66,8 @@
 	return ret;
 }
 
-#ifndef LIBC_SIGACTION
-weak_alias(__libc_sigaction,sigaction)
-libc_hidden_weak(sigaction)
-#endif
+libc_hidden_def(sigaction)
+weak_alias(sigaction,__libc_sigaction)
 
 static void
 __rt_sigreturn_stub(void)
