diff -Nur uClibc-0.9.33.2.orig/Rules.mak uClibc-0.9.33.2/Rules.mak
--- uClibc-0.9.33.2.orig/Rules.mak	2014-06-28 12:12:25.000000000 +0200
+++ uClibc-0.9.33.2/Rules.mak	2014-06-28 12:14:15.000000000 +0200
@@ -819,10 +819,7 @@
 LIBGCC_CFLAGS ?= $(CFLAGS) $(CPU_CFLAGS-y)
 $(eval $(call cache-output-var,LIBGCC,$(CC) $(LIBGCC_CFLAGS) -print-libgcc-file-name))
 $(eval $(call cache-output-var,LIBGCC_EH,$(CC) $(LIBGCC_CFLAGS) -print-file-name=libgcc_eh.a))
-# with -O0 we (e.g. lockf) might end up with references to
-# _Unwind_Resume, so pull in gcc_eh in this case..
 LIBGCC_DIR:=$(dir $(LIBGCC))
-LIBGCC += $(if $(DODEBUG),$(LIBGCC_EH))
 
 # moved from libpthread/linuxthreads
 ifeq ($(UCLIBC_CTOR_DTOR),y)
