diff -Nur linux-4.4.8.orig/arch/m68k/kernel/signal.c linux-4.4.8/arch/m68k/kernel/signal.c
--- linux-4.4.8.orig/arch/m68k/kernel/signal.c	2016-04-20 08:44:02.000000000 +0200
+++ linux-4.4.8/arch/m68k/kernel/signal.c	2016-05-08 13:32:39.067883350 +0200
@@ -595,6 +595,7 @@
 			       void __user *fp)
 {
 	int fsize = frame_extra_sizes(formatvec >> 12);
+	printk("debug: coldfire sighandler broken?..\n");
 	if (fsize < 0) {
 		/*
 		 * user process trying to return with weird frame format
