diff -Nur linux-4.1.6.orig/drivers/base/regmap/Kconfig linux-4.1.6/drivers/base/regmap/Kconfig
--- linux-4.1.6.orig/drivers/base/regmap/Kconfig	2015-08-17 05:52:51.000000000 +0200
+++ linux-4.1.6/drivers/base/regmap/Kconfig	2015-08-29 22:18:50.329683337 +0200
@@ -3,7 +3,7 @@
 # subsystems should select the appropriate symbols.
 
 config REGMAP
-	default y if (REGMAP_I2C || REGMAP_SPI || REGMAP_SPMI || REGMAP_AC97 || REGMAP_MMIO || REGMAP_IRQ)
+	default y
 	select LZO_COMPRESS
 	select LZO_DECOMPRESS
 	select IRQ_DOMAIN if REGMAP_IRQ
@@ -29,3 +29,4 @@
 
 config REGMAP_IRQ
 	bool
+	default y
