diff -Nur linux-3.5.orig//drivers/mtd/mtdpart.c linux-3.5/drivers/mtd/mtdpart.c
--- linux-3.5.orig//drivers/mtd/mtdpart.c	2012-07-21 22:58:29.000000000 +0200
+++ linux-3.5/drivers/mtd/mtdpart.c	2012-07-31 23:59:07.000000000 +0200
@@ -30,6 +30,7 @@
 #include <linux/mtd/mtd.h>
 #include <linux/mtd/partitions.h>
 #include <linux/err.h>
+#include <linux/root_dev.h>
 
 #include "mtdcore.h"
 
@@ -637,6 +638,14 @@
 		if (IS_ERR(slave))
 			return PTR_ERR(slave);
 
+		if (strcmp(parts[i].name, "rootfs") == 0) {
+			if (ROOT_DEV == 0) {
+				printk(KERN_NOTICE "mtd: partition \"rootfs\" "
+					"set to be root filesystem\n");
+					ROOT_DEV = MKDEV(MTD_BLOCK_MAJOR, i);
+			}
+		}
+
 		mutex_lock(&mtd_partitions_mutex);
 		list_add(&slave->list, &mtd_partitions);
 		mutex_unlock(&mtd_partitions_mutex);
