diff -Nur linux-3.18.5.orig/drivers/base/regmap/Kconfig linux-3.18.5/drivers/base/regmap/Kconfig
--- linux-3.18.5.orig/drivers/base/regmap/Kconfig	2015-01-30 02:41:03.000000000 +0100
+++ linux-3.18.5/drivers/base/regmap/Kconfig	2015-02-02 11:53:27.854106073 +0100
@@ -10,19 +10,19 @@
 	bool
 
 config REGMAP_I2C
-	tristate
+	bool
 	depends on I2C
 
 config REGMAP_SPI
-	tristate
+	bool
 	depends on SPI
 
 config REGMAP_SPMI
-	tristate
+	bool
 	depends on SPMI
 
 config REGMAP_MMIO
-	tristate
+	bool
 
 config REGMAP_IRQ
 	bool
