diff -Nur gcc-4.8.2.orig/gcc/config/sh/linux.h gcc-4.8.2/gcc/config/sh/linux.h
--- gcc-4.8.2.orig/gcc/config/sh/linux.h	2013-03-14 04:01:53.000000000 +0100
+++ gcc-4.8.2/gcc/config/sh/linux.h	2014-03-01 11:33:10.000000000 +0100
@@ -44,6 +44,7 @@
 #define TARGET_ASM_FILE_END file_end_indicate_exec_stack
 
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define MUSL_DYNAMIC_LINKER "/lib/ld-musl-sh.so.1"
 
 #undef SUBTARGET_LINK_EMUL_SUFFIX
 #define SUBTARGET_LINK_EMUL_SUFFIX "_linux"
