diff -Nur linux-2.6.37.orig/drivers/gpu/drm/Kconfig linux-2.6.37/drivers/gpu/drm/Kconfig
--- linux-2.6.37.orig/drivers/gpu/drm/Kconfig	2011-01-05 01:50:19.000000000 +0100
+++ linux-2.6.37/drivers/gpu/drm/Kconfig	2011-01-11 20:20:26.000000000 +0100
@@ -80,23 +80,6 @@
 	  selected, the module will be called i810.  AGP support is required
 	  for this driver to work.
 
-choice
-	prompt "Intel 830M, 845G, 852GM, 855GM, 865G"
-	depends on DRM && AGP && AGP_INTEL
-	optional
-
-config DRM_I830
-	tristate "i830 driver"
-	# BKL usage in order to avoid AB-BA deadlocks, i830 may get removed
-	depends on BKL
-	help
-	  Choose this option if you have a system that has Intel 830M, 845G,
-	  852GM, 855GM or 865G integrated graphics.  If M is selected, the
-	  module will be called i830.  AGP support is required for this driver
-	  to work. This driver is used by the older X releases X.org 6.7 and
-	  XFree86 4.3. If unsure, build this and i915 as modules and the X server
-	  will load the correct one.
-
 config DRM_I915
 	tristate "i915 driver"
 	depends on AGP_INTEL
@@ -130,8 +113,6 @@
 	  the driver to bind to PCI devices, which precludes loading things
 	  like intelfb.
 
-endchoice
-
 config DRM_MGA
 	tristate "Matrox g200/g400"
 	depends on DRM && PCI
