diff -Nur linux-2.6.39.orig/fs/Kconfig linux-2.6.39/fs/Kconfig
--- linux-2.6.39.orig/fs/Kconfig	2011-05-19 06:06:34.000000000 +0200
+++ linux-2.6.39/fs/Kconfig	2011-08-08 20:33:22.618014757 +0200
@@ -44,10 +44,10 @@
 # this symbol for ifdefs in core code.
 #
 config FS_POSIX_ACL
-	def_bool n
+	def_bool y
 
 config EXPORTFS
-	bool
+	def_bool y
 
 config FILE_LOCKING
 	bool "Enable POSIX file locking API" if EXPERT
diff -Nur linux-2.6.39.orig/fs/notify/Kconfig linux-2.6.39/fs/notify/Kconfig
--- linux-2.6.39.orig/fs/notify/Kconfig	2011-05-19 06:06:34.000000000 +0200
+++ linux-2.6.39/fs/notify/Kconfig	2011-08-08 20:33:38.328017344 +0200
@@ -1,5 +1,5 @@
 config FSNOTIFY
-	def_bool n
+	def_bool y
 
 source "fs/notify/dnotify/Kconfig"
 source "fs/notify/inotify/Kconfig"
