diff -Nur gcc-4.4.2.orig/gcc/Makefile.in gcc-4.4.2/gcc/Makefile.in
--- gcc-4.5.2.orig/gcc/Makefile.in	2010-09-02 15:05:30.000000000 +0200
+++ gcc-4.5.2/gcc/Makefile.in	2011-01-15 14:29:18.000000000 +0100
@@ -980,7 +980,8 @@ ALL_CXXFLAGS = $(T_CFLAGS) $(CXXFLAGS) $
 # Likewise.  Put INCLUDES at the beginning: this way, if some autoconf macro
 # puts -I options in CPPFLAGS, our include files in the srcdir will always
 # win against random include files in /usr/include.
-ALL_CPPFLAGS = $(INCLUDES) $(CPPFLAGS)
+#ALL_CPPFLAGS = $(INCLUDES) $(CPPFLAGS)
+ALL_CPPFLAGS = $(INCLUDES)
 
 # This is the variable to use when using $(COMPILER).
 ifneq ($(ENABLE_BUILD_WITH_CXX),yes)
