diff -Nur gcc-4.7.3.orig/gcc/config/cris/linux.h gcc-4.7.3/gcc/config/cris/linux.h
--- gcc-4.7.3.orig/gcc/config/cris/linux.h	2011-04-11 20:46:05.000000000 +0200
+++ gcc-4.7.3/gcc/config/cris/linux.h	2013-10-11 10:01:51.000000000 +0200
@@ -104,6 +104,7 @@
 #define CRIS_DEFAULT_CPU_VERSION CRIS_CPU_NG
 
 #define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define MUSL_DYNAMIC_LINKER "/lib/ld-musl-cris.1"
 
 #undef CRIS_LINK_SUBTARGET_SPEC
 #define CRIS_LINK_SUBTARGET_SPEC \
