diff -Nur binutils-2.25.1.orig/bfd/elf32-m68k.c binutils-2.25.1/bfd/elf32-m68k.c
--- binutils-2.25.1.orig/bfd/elf32-m68k.c	2015-07-21 10:20:58.000000000 +0200
+++ binutils-2.25.1/bfd/elf32-m68k.c	2016-10-30 17:32:01.125211066 +0100
@@ -2637,7 +2637,6 @@
 	  if (sgot == NULL)
 	    {
 	      sgot = bfd_get_linker_section (dynobj, ".got");
-	      BFD_ASSERT (sgot != NULL);
 	    }
 
 	  if (srelgot == NULL
