diff -Nur binutils-2.25.1.orig/ld/configure.tgt binutils-2.25.1/ld/configure.tgt
--- binutils-2.25.1.orig/ld/configure.tgt	2014-10-14 09:32:04.000000000 +0200
+++ binutils-2.25.1/ld/configure.tgt	2016-03-10 00:45:31.000000000 +0100
@@ -525,8 +525,8 @@
 nds32*be-*-elf*)	targ_emul=nds32belf
 			targ_extra_emuls="nds32elf nds32elf16m nds32belf16m"
 			;;
-nds32*le-*-linux-gnu*)	targ_emul=nds32elf_linux ;;
-nds32*be-*-linux-gnu*)	targ_emul=nds32belf_linux ;;
+nds32*le-*-linux*)	targ_emul=nds32elf_linux ;;
+nds32*be-*-linux*)	targ_emul=nds32belf_linux ;;
 nios2*-*-linux*)	targ_emul=nios2linux ;;
 nios2*-*-*)		targ_emul=nios2elf ;;
 ns32k-pc532-mach* | ns32k-pc532-ux*)  targ_emul=pc532macha ;;
