diff -Nur linux-3.11.5.orig/kernel/pid.c linux-3.11.5/kernel/pid.c
--- linux-3.11.5.orig/kernel/pid.c	2013-10-14 03:14:45.000000000 +0200
+++ linux-3.11.5/kernel/pid.c	2013-10-29 15:37:02.000000000 +0100
@@ -450,6 +450,7 @@
 {
 	return find_task_by_pid_ns(vnr, task_active_pid_ns(current));
 }
+EXPORT_SYMBOL(find_task_by_vpid);
 
 struct pid *get_task_pid(struct task_struct *task, enum pid_type type)
 {
