diff -Nur linux-3.2.66.orig/arch/sparc/boot/Makefile linux-3.2.66/arch/sparc/boot/Makefile
--- linux-3.2.66.orig/arch/sparc/boot/Makefile	2014-12-31 19:27:54.000000000 -0600
+++ linux-3.2.66/arch/sparc/boot/Makefile	2015-02-17 13:21:38.269400789 -0600
@@ -75,9 +75,6 @@
 	$(call if_changed,gzip)
 	@echo '  kernel: $@ is ready'
 
-$(obj)/vmlinux.aout: vmlinux FORCE
-	$(call if_changed,elftoaout)
-	@echo '  kernel: $@ is ready'
 else
 
 # The following lines make a readable image for U-Boot.
@@ -110,6 +107,10 @@
 
 endif
 
+$(obj)/vmlinux.aout: $(obj)/image FORCE
+	$(call if_changed,elftoaout)
+	@echo '  kernel: $@ is ready'
+
 $(obj)/tftpboot.img: $(obj)/image $(obj)/piggyback System.map $(ROOT_IMG) FORCE
 	$(call if_changed,elftoaout)
 	$(call if_changed,piggy)
