diff -Nur binutils-nds32-binutils-2.30-branch-open.orig/ld/configure.tgt binutils-nds32-binutils-2.30-branch-open/ld/configure.tgt
--- binutils-nds32-binutils-2.30-branch-open.orig/ld/configure.tgt	2018-05-10 22:03:15.000000000 +0200
+++ binutils-nds32-binutils-2.30-branch-open/ld/configure.tgt	2018-05-12 08:44:24.000000000 +0200
@@ -583,8 +583,8 @@
 nds32*be-*-elf*)	targ_emul=nds32belf
 			targ_extra_emuls="nds32elf nds32elf16m nds32belf16m"
 			;;
-nds32*le-*-linux-gnu*)	targ_emul=nds32elf_linux ;;
-nds32*be-*-linux-gnu*)	targ_emul=nds32belf_linux ;;
+nds32*le-*-linux*)	targ_emul=nds32elf_linux ;;
+nds32*be-*-linux*)	targ_emul=nds32belf_linux ;;
 nios2*-*-linux*)	targ_emul=nios2linux ;;
 nios2*-*-*)		targ_emul=nios2elf ;;
 ns32k-pc532-mach* | ns32k-pc532-ux*)  targ_emul=pc532macha ;;
