diff -Nur gcc-4.2.4.orig/gcc/toplev.h gcc-4.2.4/gcc/toplev.h
--- gcc-4.2.4.orig/gcc/toplev.h	2007-09-01 17:28:30.000000000 +0200
+++ gcc-4.2.4/gcc/toplev.h	2017-06-30 22:54:25.600188422 +0200
@@ -150,38 +150,6 @@
 /* Return true iff flags are set as if -ffast-math.  */
 extern bool fast_math_flags_set_p	(void);
 
-/* Return log2, or -1 if not exact.  */
-extern int exact_log2                  (unsigned HOST_WIDE_INT);
-
-/* Return floor of log2, with -1 for zero.  */
-extern int floor_log2                  (unsigned HOST_WIDE_INT);
-
-/* Inline versions of the above for speed.  */
-#if GCC_VERSION >= 3004
-# if HOST_BITS_PER_WIDE_INT == HOST_BITS_PER_LONG
-#  define CLZ_HWI __builtin_clzl
-#  define CTZ_HWI __builtin_ctzl
-# elif HOST_BITS_PER_WIDE_INT == HOST_BITS_PER_LONGLONG
-#  define CLZ_HWI __builtin_clzll
-#  define CTZ_HWI __builtin_ctzll
-# else
-#  define CLZ_HWI __builtin_clz
-#  define CTZ_HWI __builtin_ctz
-# endif
-
-extern inline int
-floor_log2 (unsigned HOST_WIDE_INT x)
-{
-  return x ? HOST_BITS_PER_WIDE_INT - 1 - (int) CLZ_HWI (x) : -1;
-}
-
-extern inline int
-exact_log2 (unsigned HOST_WIDE_INT x)
-{
-  return x == (x & -x) && x ? (int) CTZ_HWI (x) : -1;
-}
-#endif /* GCC_VERSION >= 3004 */
-
 /* Functions used to get and set GCC's notion of in what directory
    compilation was started.  */
 
