diff -Nur newlib-2.2.0-1.orig/etc/Makefile.in newlib-2.2.0-1/etc/Makefile.in
--- newlib-2.2.0-1.orig/etc/Makefile.in	2010-11-20 20:37:57.000000000 +0100
+++ newlib-2.2.0-1/etc/Makefile.in	2015-12-03 14:23:44.043820971 +0100
@@ -63,8 +63,8 @@
 PDFFILES = standards.pdf configure.pdf
 HTMLFILES = standards.html configure.html
 
-all: info
-install install-strip: install-info
+all:
+install install-strip:
 
 uninstall:
 
