--- watchdog-5.2.4.orig.orig/src/fstab.c	2003-01-28 07:47:38.000000000 +0100
+++ watchdog-5.2.4.orig/src/fstab.c	2008-10-16 12:56:40.000000000 +0200
@@ -237,7 +237,7 @@ static int old_lockfile = 1;
 /* Ensure that the lock is released if we are interrupted.  */
 static void
 handler (int sig) {
-     die (EX_USER, "%s", sys_siglist[sig]);
+     die (EX_USER, "%s", "sys_siglist[sig]");
 }
 
 static void
