Index: os/linux/ccconfig.h
===================================================================
RCS file: /cvsroot/pcc/os/linux/ccconfig.h,v
retrieving revision 1.19
diff -u -p -r1.19 ccconfig.h
--- a/os/linux/ccconfig.h	9 Nov 2010 08:40:50 -0000	1.19
+++ b/os/linux/ccconfig.h	2 Apr 2011 14:19:17 -0000
@@ -67,6 +67,15 @@
 #error defines for arch missing
 #endif
 
+#ifdef ADK_FOR_UCLIBC
+#undef DYNLINKER
+#define DYNLINKER { "-dynamic-linker", "/lib/ld-uClibc.so.0", NULL }
+#elif defined(ADK_FOR_GLIBC)
+/* see above */
+#else
+#error What now?
+#endif
+
 #ifndef LIBDIR
 #define LIBDIR "/usr/lib/"
 #endif
