--- a/configure	Fri Apr  1 13:58:07 2011
+++ b/configure	Sat Apr  2 19:21:15 2011
@@ -672,6 +672,8 @@ ac_ct_CC
 EXEEXT
 OBJEXT
 CC_FOR_BUILD
+CPPFLAGS_FOR_BUILD
+CFLAGS_FOR_BUILD
 CPP
 GREP
 EGREP
@@ -3292,6 +3294,8 @@ else
 echo "${ECHO_T}not cross compiling" >&6; }
         CC_FOR_BUILD=${CC-cc}
 
+	test -n "$CPPFLAGS_FOR_BUILD" || CPPFLAGS_FOR_BUILD=$CPPFLAGS
+	test -n "$CFLAGS_FOR_BUILD" || CFLAGS_FOR_BUILD=$CFLAGS
 fi
 
 
@@ -5765,6 +5769,8 @@ ac_ct_CC!$ac_ct_CC$ac_delim
 EXEEXT!$EXEEXT$ac_delim
 OBJEXT!$OBJEXT$ac_delim
 CC_FOR_BUILD!$CC_FOR_BUILD$ac_delim
+CPPFLAGS_FOR_BUILD!$CPPFLAGS_FOR_BUILD$ac_delim
+CFLAGS_FOR_BUILD!$CFLAGS_FOR_BUILD$ac_delim
 CPP!$CPP$ac_delim
 GREP!$GREP$ac_delim
 EGREP!$EGREP$ac_delim
@@ -5787,7 +5793,7 @@ LIBOBJS!$LIBOBJS$ac_delim
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 78; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 80; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
