diff -Nur linux-3.15-rc5.orig/drivers/tty/serial/serial_core.c linux-3.15-rc5/drivers/tty/serial/serial_core.c
--- linux-3.15-rc5.orig/drivers/tty/serial/serial_core.c	2014-05-09 22:10:52.000000000 +0200
+++ linux-3.15-rc5/drivers/tty/serial/serial_core.c	2014-05-20 10:30:52.000000000 +0200
@@ -2686,9 +2686,10 @@
 
 	/*
 	 * If the port is used as a console, unregister it
-	 */
+ 	 * DISABLE: breaks serial console on rb532
 	if (uart_console(uport))
 		unregister_console(uport->cons);
+	 */
 
 	/*
 	 * Free the port IO and memory resources, if any.
