From 05ca0b1b73f368564aef8cc4763e96a218cfefeb Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbrodkorb@conet.de>
Date: Mon, 2 Feb 2015 09:51:56 +0100
Subject: [PATCH 2/2] allow to control volume for each channel

Signed-off-by: Waldemar Brodkorb <wbrodkorb@conet.de>
---
 sound/arm/bcm2835-ctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/arm/bcm2835-ctl.c b/sound/arm/bcm2835-ctl.c
index aad905f..12e77f5 100755
--- a/sound/arm/bcm2835-ctl.c
+++ b/sound/arm/bcm2835-ctl.c
@@ -45,7 +45,7 @@ static int snd_bcm2835_ctl_info(struct snd_kcontrol *kcontrol,
 	audio_info(" ... IN\n");
 	if (kcontrol->private_value == PCM_PLAYBACK_VOLUME) {
 		uinfo->type = SNDRV_CTL_ELEM_TYPE_INTEGER;
-		uinfo->count = 1;
+		uinfo->count = 2;
 		uinfo->value.integer.min = CTRL_VOL_MIN;
 		uinfo->value.integer.max = CTRL_VOL_MAX;      /* 2303 */
 	} else if (kcontrol->private_value == PCM_PLAYBACK_MUTE) {
-- 
1.9.1

