diff -Nur linux-3.10.30.orig/scripts/Makefile.headersinst linux-3.10.30/scripts/Makefile.headersinst
--- linux-3.10.30.orig/scripts/Makefile.headersinst	2014-02-13 22:48:15.000000000 +0100
+++ linux-3.10.30/scripts/Makefile.headersinst	2014-03-09 09:47:17.000000000 +0100
@@ -99,7 +99,6 @@
 
 targets += $(install-file)
 $(install-file): scripts/headers_install.sh $(input-files) FORCE
-	$(if $(unwanted),$(call cmd,remove),)
 	$(if $(wildcard $(dir $@)),,$(shell mkdir -p $(dir $@)))
 	$(call if_changed,install)
 
