diff -Nur linux-2.6.32.orig/init/main.c linux-2.6.32/init/main.c
--- linux-2.6.32.orig/init/main.c	2009-12-03 04:51:21.000000000 +0100
+++ linux-2.6.32/init/main.c	2010-01-31 11:01:41.154334301 +0100
@@ -814,6 +814,8 @@
 	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
 		printk(KERN_WARNING "Warning: unable to open an initial console.\n");
 
+	printk(KERN_WARNING "Starting Linux (build with OpenADK).\n");
+
 	(void) sys_dup(0);
 	(void) sys_dup(0);
 
@@ -836,6 +838,7 @@
 		printk(KERN_WARNING "Failed to execute %s.  Attempting "
 					"defaults...\n", execute_command);
 	}
+	run_init_process("/init");
 	run_init_process("/sbin/init");
 	run_init_process("/etc/init");
 	run_init_process("/bin/init");
