diff -Nur gcc-4.7.2.orig/gcc/config/i386/linux64.h gcc-4.7.2/gcc/config/i386/linux64.h
--- gcc-4.7.2.orig/gcc/config/i386/linux64.h	2011-07-07 17:38:34.000000000 +0200
+++ gcc-4.7.2/gcc/config/i386/linux64.h	2012-09-22 12:48:17.000000000 +0200
@@ -29,5 +29,5 @@
 #define GNU_USER_LINK_EMULATIONX32 "elf32_x86_64"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
-#define GLIBC_DYNAMIC_LINKERX32 "/libx32/ld-linux-x32.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKERX32 "/lib/ld-linux-so.2"
diff -Nur gcc-4.7.2.orig/gcc/config/i386/t-linux64 gcc-4.7.2/gcc/config/i386/t-linux64
--- gcc-4.7.2.orig/gcc/config/i386/t-linux64	2011-11-02 16:23:48.000000000 +0100
+++ gcc-4.7.2/gcc/config/i386/t-linux64	2012-09-22 12:46:07.000000000 +0200
@@ -34,6 +34,6 @@
 comma=,
 MULTILIB_OPTIONS    = $(subst $(comma),/,$(TM_MULTILIB_CONFIG))
 MULTILIB_DIRNAMES   = $(patsubst m%, %, $(subst /, ,$(MULTILIB_OPTIONS)))
-MULTILIB_OSDIRNAMES = m64=../lib64
-MULTILIB_OSDIRNAMES+= m32=$(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
-MULTILIB_OSDIRNAMES+= mx32=../libx32
+MULTILIB_OSDIRNAMES = m64=../lib
+MULTILIB_OSDIRNAMES+= m32=../lib
+MULTILIB_OSDIRNAMES+= mx32=../lib
diff -Nur gcc-4.7.2.orig/gcc/config/mips/linux64.h gcc-4.7.2/gcc/config/mips/linux64.h
--- gcc-4.7.2.orig/gcc/config/mips/linux64.h	2011-07-19 20:00:27.000000000 +0200
+++ gcc-4.7.2/gcc/config/mips/linux64.h	2012-09-22 12:44:16.000000000 +0200
@@ -24,9 +24,9 @@
 #define GNU_USER_LINK_EMULATIONN32 "elf32%{EB:b}%{EL:l}tsmipn32"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERN32 "/lib32/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKERN32 "/lib32/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERN32 "/lib/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKERN32 "/lib/ld-uClibc.so.0"
 #define BIONIC_DYNAMIC_LINKERN32 "/system/bin/linker32"
 #define GNU_USER_DYNAMIC_LINKERN32 \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKERN32, UCLIBC_DYNAMIC_LINKERN32, \
diff -Nur gcc-4.7.2.orig/gcc/config/mips/t-linux64 gcc-4.7.2/gcc/config/mips/t-linux64
--- gcc-4.7.2.orig/gcc/config/mips/t-linux64	2011-11-02 15:33:56.000000000 +0100
+++ gcc-4.7.2/gcc/config/mips/t-linux64	2012-09-22 12:48:39.000000000 +0200
@@ -18,4 +18,4 @@
 
 MULTILIB_OPTIONS = mabi=n32/mabi=32/mabi=64
 MULTILIB_DIRNAMES = n32 32 64
-MULTILIB_OSDIRNAMES = ../lib32 ../lib ../lib64
+MULTILIB_OSDIRNAMES = ../lib ../lib ../lib
diff -Nur gcc-4.7.2.orig/gcc/config/rs6000/linux64.h gcc-4.7.2/gcc/config/rs6000/linux64.h
--- gcc-4.7.2.orig/gcc/config/rs6000/linux64.h	2012-03-12 17:16:51.000000000 +0100
+++ gcc-4.7.2/gcc/config/rs6000/linux64.h	2012-09-22 12:44:16.000000000 +0200
@@ -359,7 +359,7 @@
 #define LINK_OS_DEFAULT_SPEC "%(link_os_linux)"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld64.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
 #define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
 #if DEFAULT_LIBC == LIBC_UCLIBC
diff -Nur gcc-4.7.2.orig/gcc/config/rs6000/t-linux64 gcc-4.7.2/gcc/config/rs6000/t-linux64
--- gcc-4.7.2.orig/gcc/config/rs6000/t-linux64	2011-11-02 16:23:48.000000000 +0100
+++ gcc-4.7.2/gcc/config/rs6000/t-linux64	2012-09-22 12:44:16.000000000 +0200
@@ -31,5 +31,5 @@
 MULTILIB_EXTRA_OPTS     = fPIC mstrict-align
 MULTILIB_EXCEPTIONS     = m64/msoft-float
 MULTILIB_EXCLUSIONS     = m64/!m32/msoft-float
-MULTILIB_OSDIRNAMES	= ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib) nof
+MULTILIB_OSDIRNAMES	= ../lib $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib) nof
 MULTILIB_MATCHES        = $(MULTILIB_MATCHES_FLOAT)
diff -Nur gcc-4.7.2.orig/gcc/config/sparc/linux64.h gcc-4.7.2/gcc/config/sparc/linux64.h
--- gcc-4.7.2.orig/gcc/config/sparc/linux64.h	2011-11-12 02:05:07.000000000 +0100
+++ gcc-4.7.2/gcc/config/sparc/linux64.h	2012-09-22 12:44:16.000000000 +0200
@@ -94,7 +94,7 @@
    done.  */
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
diff -Nur gcc-4.7.2.orig/gcc/config/sparc/t-linux64 gcc-4.7.2/gcc/config/sparc/t-linux64
--- gcc-4.7.2.orig/gcc/config/sparc/t-linux64	2011-11-02 16:23:48.000000000 +0100
+++ gcc-4.7.2/gcc/config/sparc/t-linux64	2012-09-22 12:49:43.000000000 +0200
@@ -26,4 +26,4 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+MULTILIB_OSDIRNAMES = ../lib ../lib
