diff -Nur gcc-4.4.0.orig/gcc/config/i386/linux64.h gcc-4.4.0/gcc/config/i386/linux64.h
--- gcc-4.4.0.orig/gcc/config/i386/linux64.h	2009-04-10 01:23:07.000000000 +0200
+++ gcc-4.4.0/gcc/config/i386/linux64.h	2009-06-02 22:38:29.054028385 +0200
@@ -59,7 +59,7 @@
    done.  */
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux.so.2"
 
 #undef ASM_SPEC
 #define ASM_SPEC "%{v:-V} %{Qy:} %{!Qn:-Qy} %{n} %{T} %{Ym,*} %{Yd,*} \
diff -Nur gcc-4.4.0.orig/gcc/config/i386/t-linux64 gcc-4.4.0/gcc/config/i386/t-linux64
--- gcc-4.4.0.orig/gcc/config/i386/t-linux64	2008-07-02 17:59:19.000000000 +0200
+++ gcc-4.4.0/gcc/config/i386/t-linux64	2009-06-02 22:40:00.800873792 +0200
@@ -7,7 +7,7 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+MULTILIB_OSDIRNAMES = ../lib $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
diff -Nur gcc-4.4.0.orig/gcc/config/mips/linux64.h gcc-4.4.0/gcc/config/mips/linux64.h
--- gcc-4.4.0.orig/gcc/config/mips/linux64.h	2009-02-20 16:20:38.000000000 +0100
+++ gcc-4.4.0/gcc/config/mips/linux64.h	2009-06-02 22:40:46.756972667 +0200
@@ -36,7 +36,7 @@
   %{profile:-lc_p} %{!profile:-lc}}"
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld.so.1"
 #define GLIBC_DYNAMIC_LINKERN32 "/lib32/ld.so.1"
 #define UCLIBC_DYNAMIC_LINKERN32 "/lib32/ld-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKERN32 \
diff -Nur gcc-4.4.0.orig/gcc/config/mips/t-linux64 gcc-4.4.0/gcc/config/mips/t-linux64
--- gcc-4.4.0.orig/gcc/config/mips/t-linux64	2006-06-06 14:51:24.000000000 +0200
+++ gcc-4.4.0/gcc/config/mips/t-linux64	2009-06-02 22:35:11.634464615 +0200
@@ -1,6 +1,6 @@
 MULTILIB_OPTIONS = mabi=n32/mabi=32/mabi=64
 MULTILIB_DIRNAMES = n32 32 64
-MULTILIB_OSDIRNAMES = ../lib32 ../lib ../lib64
+MULTILIB_OSDIRNAMES = ../lib ../lib ../lib
 
 EXTRA_MULTILIB_PARTS=crtbegin.o crtend.o crtbeginS.o crtendS.o crtbeginT.o
 
