diff -Nur gcc-5.3.0.orig/libgcc/config.host gcc-5.3.0/libgcc/config.host
--- gcc-5.3.0.orig/libgcc/config.host	2015-10-01 14:01:18.000000000 +0200
+++ gcc-5.3.0/libgcc/config.host	2016-05-30 00:20:39.299872144 +0200
@@ -948,6 +948,10 @@
 		;;
 	esac
 	;;
+nios2-*-linux-uclibc*)
+	tmake_file="$tmake_file nios2/t-nios2 nios2/t-linux t-libgcc-pic t-slibgcc-libgcc t-softfp-sfdf t-softfp"
+	md_unwind_header=nios2/linux-unwind.h
+	;;
 nios2-*-linux*)
 	tmake_file="$tmake_file nios2/t-nios2 nios2/t-linux t-libgcc-pic t-slibgcc-libgcc"
 	md_unwind_header=nios2/linux-unwind.h
