diff -Nur linux-3.15.orig/drivers/base/regmap/Kconfig linux-3.15/drivers/base/regmap/Kconfig
--- linux-3.15.orig/drivers/base/regmap/Kconfig	2014-06-08 20:19:54.000000000 +0200
+++ linux-3.15/drivers/base/regmap/Kconfig	2014-06-20 10:33:33.450143320 +0200
@@ -10,16 +10,16 @@
 	bool
 
 config REGMAP_I2C
-	tristate
+	boolean
 
 config REGMAP_SPI
-	tristate
+	boolean
 
 config REGMAP_SPMI
-	tristate
+	boolean
 
 config REGMAP_MMIO
-	tristate
+	boolean
 
 config REGMAP_IRQ
 	bool
