diff -Nur gcc-6.1.0.orig/gcc/config/frv/frv.h gcc-6.1.0/gcc/config/frv/frv.h
--- gcc-6.1.0.orig/gcc/config/frv/frv.h	2016-01-04 15:30:50.000000000 +0100
+++ gcc-6.1.0/gcc/config/frv/frv.h	2016-08-16 13:31:32.652520944 +0200
@@ -1350,11 +1350,8 @@
 #define TRAMPOLINE_TEMPLATE_NAME "__trampoline_template"
 #endif
 
-#define Twrite _write
-
 #if ! __FRV_FDPIC__
 #define TRANSFER_FROM_TRAMPOLINE					\
-extern int Twrite (int, const void *, unsigned);			\
 									\
 void									\
 __trampoline_setup (short * addr, int size, int fnaddr, int sc)		\
@@ -1366,8 +1363,6 @@
 									\
   if (size < 20)							\
     {									\
-      Twrite (2, "__trampoline_setup bad size\n",			\
-	      sizeof ("__trampoline_setup bad size\n") - 1);		\
       exit (-1);							\
     }									\
 									\
@@ -1397,7 +1392,6 @@
 	"\tjmpl @(gr0,gr6)\n");
 #else
 #define TRANSFER_FROM_TRAMPOLINE					\
-extern int Twrite (int, const void *, unsigned);			\
 									\
 void									\
 __trampoline_setup (addr, size, fnaddr, sc)				\
@@ -1414,8 +1408,6 @@
 									\
   if (size != 32)							\
     {									\
-      Twrite (2, "__trampoline_setup bad size\n",			\
-	      sizeof ("__trampoline_setup bad size\n") - 1);		\
       exit (-1);							\
     }									\
 									\
diff -Nur gcc-6.1.0.orig/gcc/config/frv/linux.h gcc-6.1.0/gcc/config/frv/linux.h
--- gcc-6.1.0.orig/gcc/config/frv/linux.h	2016-01-04 15:30:50.000000000 +0100
+++ gcc-6.1.0/gcc/config/frv/linux.h	2016-08-16 13:31:06.763517719 +0200
@@ -68,7 +68,4 @@
 
 #undef INVOKE__main
 
-#undef Twrite
-#define Twrite __write
-
 #endif /* __FRV_LINUX_H__ */
