From:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65649

diff -Nur gcc-6.3.0.orig/gcc/config/microblaze/microblaze.c gcc-6.3.0/gcc/config/microblaze/microblaze.c
--- gcc-6.3.0.orig/gcc/config/microblaze/microblaze.c	2016-01-21 18:15:52.000000000 +0100
+++ gcc-6.3.0/gcc/config/microblaze/microblaze.c	2017-01-03 21:54:26.462805169 +0100
@@ -2364,7 +2364,7 @@
 	  unsigned long value_long;
 	  REAL_VALUE_TO_TARGET_SINGLE (*CONST_DOUBLE_REAL_VALUE (op),
 				       value_long);
-	  fprintf (file, HOST_WIDE_INT_PRINT_HEX, value_long);
+	  fprintf (file, "0x%lx", value_long);
 	}
       else
 	{
@@ -2423,7 +2423,7 @@
       print_operand_address (file, XEXP (op, 0));
     }
   else if (letter == 'm')
-    fprintf (file, HOST_WIDE_INT_PRINT_DEC, (1L << INTVAL (op)));
+    fprintf (file, "%ld", (1L << INTVAL (op)));
   else
     output_addr_const (file, op);
 }
