diff -Nur classpath-0.98.orig//lib/Makefile.in classpath-0.98/lib/Makefile.in
--- classpath-0.99.orig/lib/Makefile.in	2012-03-09 00:54:14.000000000 +0100
+++ classpath-0.99/lib/Makefile.in	2014-08-24 21:21:19.000000000 +0200
@@ -305,7 +305,7 @@ top_srcdir = @top_srcdir@
 uudecode = @uudecode@
 vm_classes = @vm_classes@
 JAVA_DEPEND = java.dep
-compile_classpath = $(vm_classes):$(top_srcdir):$(top_srcdir)/external/w3c_dom:$(top_srcdir)/external/sax:$(top_srcdir)/external/relaxngDatatype:$(top_srcdir)/external/jsr166:.:$(PATH_TO_GLIBJ_ZIP):$(PATH_TO_ESCHER)
+compile_classpath = $(vm_classes) -cp $(top_srcdir) -cp $(top_srcdir)/external/w3c_dom -cp $(top_srcdir)/external/sax -cp $(top_srcdir)/external/relaxngDatatype -cp $(top_srcdir)/external/jsr166 -cp . -cp $(PATH_TO_GLIBJ_ZIP) -cp $(PATH_TO_ESCHER)
 @GCJ_JAVAC_FALSE@JCOMPILER = $(JAVAC) $(JAVACFLAGS) $(JAVAC_MEM_OPT) -source 1.5 -target 1.5 -bootclasspath '' -classpath $(compile_classpath) -d . @classes
 
 # handling source to bytecode compiler programs like gcj, jikes  and kjc
