From 8cbc2ee92ec6dbed4a806cedffc6919b6b90275b Mon Sep 17 00:00:00 2001
From: Phil Sutter <phil@nwl.cc>
Date: Tue, 3 Jun 2014 00:32:22 +0200
Subject: [PATCH] rb4xx_nand: add partition for cfgfs

---
 drivers/mtd/nand/rb4xx_nand.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/mtd/nand/rb4xx_nand.c b/drivers/mtd/nand/rb4xx_nand.c
index 5b9841b..603d001 100644
--- a/drivers/mtd/nand/rb4xx_nand.c
+++ b/drivers/mtd/nand/rb4xx_nand.c
@@ -65,6 +65,11 @@ static struct mtd_partition rb4xx_nand_partitions[] = {
 		.size	= (4 * 1024 * 1024) - (256 * 1024),
 	},
 	{
+		.name	= "cfgfs",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x400000,
+	},
+	{
 		.name	= "rootfs",
 		.offset	= MTDPART_OFS_NXTBLK,
 		.size	= MTDPART_SIZ_FULL,
-- 
1.8.5.3

