diff -Nur linux-or1k-fixes-4.9.orig/arch/openrisc/boot/dts/or1ksim.dts linux-or1k-fixes-4.9/arch/openrisc/boot/dts/or1ksim.dts
--- linux-or1k-fixes-4.9.orig/arch/openrisc/boot/dts/or1ksim.dts	2016-11-17 21:49:29.000000000 +0100
+++ linux-or1k-fixes-4.9/arch/openrisc/boot/dts/or1ksim.dts	2016-11-21 14:36:14.674136327 +0100
@@ -11,7 +11,7 @@
 
 	memory@0 {
 		device_type = "memory";
-		reg = <0x00000000 0x02000000>;
+		reg = <0x00000000 0x04000000>;
 	};
 
 	cpus {
