diff -Nur linux-2.6.39.orig/arch/cris/include/asm/thread_info.h linux-2.6.39/arch/cris/include/asm/thread_info.h
--- linux-2.6.39.orig/arch/cris/include/asm/thread_info.h	2011-05-19 06:06:34.000000000 +0200
+++ linux-2.6.39/arch/cris/include/asm/thread_info.h	2011-06-28 13:07:09.712682140 +0200
@@ -68,7 +68,7 @@
 #define init_thread_info	(init_thread_union.thread_info)
 
 /* thread information allocation */
-#define alloc_thread_info(tsk, node) ((struct thread_info *) __get_free_pages(GFP_KERNEL,1))
+#define alloc_thread_info_node(tsk, node) ((struct thread_info *) __get_free_pages(GFP_KERNEL,1))
 #define free_thread_info(ti) free_pages((unsigned long) (ti), 1)
 
 #endif /* !__ASSEMBLY__ */
