diff -Nur linux-4.1.10.orig/scripts/Makefile.headersinst linux-4.1.10/scripts/Makefile.headersinst
--- linux-4.1.10.orig/scripts/Makefile.headersinst	2015-10-03 13:49:38.000000000 +0200
+++ linux-4.1.10/scripts/Makefile.headersinst	2015-10-15 11:23:35.000000000 +0200
@@ -107,7 +107,6 @@
 
 targets += $(install-file)
 $(install-file): scripts/headers_install.sh $(input-files1) $(input-files2) $(input-files3) FORCE
-	$(if $(unwanted),$(call cmd,remove),)
 	$(if $(wildcard $(dir $@)),,$(shell mkdir -p $(dir $@)))
 	$(call if_changed,install)
 
