diff -Nur glibc-2.7.orig/configure glibc-2.7/configure
--- glibc-2.7.orig/configure	2007-10-18 13:22:23.000000000 +0200
+++ glibc-2.7/configure	2008-11-29 23:48:57.461566623 +0100
@@ -5062,8 +5062,12 @@
 # header directory and add that to the list.  NOTE: Only does the right
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
-  ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  SYSINCLUDES=-nostdinc
+  for d in include include-fixed; do
+  	i=`$CC -print-file-name="$d"` && test "x$i" != "x$d" &&
+	SYSINCLUDES="$SYSINCLUDES -isystem $i"
+  done
+  SYSINCLUDES="$SYSINCLUDES \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&5` &&
