diff -Nur glibc-2.7.orig/sysdeps/i386/i586/memcpy_chk.S glibc-2.7/sysdeps/i386/i586/memcpy_chk.S
--- glibc-2.7.orig/sysdeps/i386/i586/memcpy_chk.S	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.7/sysdeps/i386/i586/memcpy_chk.S	2008-09-17 12:22:02.000000000 +0200
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/memcpy_chk.S>
diff -Nur glibc-2.7.orig/sysdeps/i386/i586/mempcpy_chk.S glibc-2.7/sysdeps/i386/i586/mempcpy_chk.S
--- glibc-2.7.orig/sysdeps/i386/i586/mempcpy_chk.S	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.7/sysdeps/i386/i586/mempcpy_chk.S	2008-09-17 12:22:02.000000000 +0200
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/mempcpy_chk.S>
diff -Nur glibc-2.7.orig/sysdeps/i386/i586/memset_chk.S glibc-2.7/sysdeps/i386/i586/memset_chk.S
--- glibc-2.7.orig/sysdeps/i386/i586/memset_chk.S	1970-01-01 01:00:00.000000000 +0100
+++ glibc-2.7/sysdeps/i386/i586/memset_chk.S	2008-09-17 12:22:02.000000000 +0200
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/memset_chk.S>
