diff -Nur glibc-2.10-current.orig/ports/sysdeps/mips/mips64/n32/Makefile glibc-2.10-current/ports/sysdeps/mips/mips64/n32/Makefile
--- glibc-2.10-current.orig/ports/sysdeps/mips/mips64/n32/Makefile	2009-05-26 22:53:27.000000000 +0200
+++ glibc-2.10-current/ports/sysdeps/mips/mips64/n32/Makefile	2009-05-27 02:23:06.000000000 +0200
@@ -1,6 +1,2 @@
 # `long double' is a distinct type we support.
 long-double-fcts = yes
-
-ifeq ($(filter -mabi=n32,$(CC)),)
-CC += -mabi=n32
-endif
diff -Nur glibc-2.10-current.orig/ports/sysdeps/mips/mips64/n64/Makefile glibc-2.10-current/ports/sysdeps/mips/mips64/n64/Makefile
--- glibc-2.10-current.orig/ports/sysdeps/mips/mips64/n64/Makefile	2009-05-26 22:53:27.000000000 +0200
+++ glibc-2.10-current/ports/sysdeps/mips/mips64/n64/Makefile	2009-05-27 02:20:00.000000000 +0200
@@ -1,6 +1,2 @@
 # `long double' is a distinct type we support.
 long-double-fcts = yes
-
-ifeq ($(filter -mabi=64,$(CC)),)
-CC += -mabi=64
-endif
diff -Nur glibc-2.10-current.orig/ports/sysdeps/mips/preconfigure glibc-2.10-current/ports/sysdeps/mips/preconfigure
--- glibc-2.10-current.orig/ports/sysdeps/mips/preconfigure	2009-05-26 22:53:27.000000000 +0200
+++ glibc-2.10-current/ports/sysdeps/mips/preconfigure	2009-05-27 02:20:14.000000000 +0200
@@ -22,7 +22,7 @@
 		if test $mips_config_abi != $mips_cc_abi; then
 		  # This won't make it to config.make, but we want to
 		  # set this in case configure tests depend on it.
-		  CPPFLAGS="$CPPFLAGS -mabi=$mips_config_abi"
+		  CPPFLAGS="$CPPFLAGS"
 		fi
 		;;
 mips*)		base_machine=mips machine=mips/mips32/$machine ;;
diff -Nur glibc-2.10-current.orig/ports/sysdeps/unix/sysv/linux/mips/configure glibc-2.10-current/ports/sysdeps/unix/sysv/linux/mips/configure
--- glibc-2.10-current.orig/ports/sysdeps/unix/sysv/linux/mips/configure	2009-05-26 22:53:29.000000000 +0200
+++ glibc-2.10-current/ports/sysdeps/unix/sysv/linux/mips/configure	2009-05-27 20:34:25.000000000 +0200
@@ -84,22 +84,6 @@
   # and libc_cv_localedir.
   test -n "$libc_cv_slibdir" || \
   case $machine in
-  mips/mips64/n64/* )
-    libc_cv_slibdir="/lib64"
-    if test "$libdir" = '${exec_prefix}/lib'; then
-      libdir='${exec_prefix}/lib64';
-      # Locale data can be shared between 32bit and 64bit libraries
-      libc_cv_localedir='${exec_prefix}/lib/locale'
-    fi
-    ;;
-  mips/mips64/n32/* )
-    libc_cv_slibdir="/lib32"
-    if test "$libdir" = '${exec_prefix}/lib'; then
-      libdir='${exec_prefix}/lib32';
-      # Locale data can be shared between 32bit and 64bit libraries
-      libc_cv_localedir='${exec_prefix}/lib/locale'
-    fi
-    ;;
   *)
     libc_cv_slibdir="/lib"
     ;;
