diff -Nur glibc-2.19.orig/configure glibc-2.19/configure
--- glibc-2.19.orig/configure	2014-02-07 10:04:38.000000000 +0100
+++ glibc-2.19/configure	2014-04-16 23:11:01.000000000 +0200
@@ -6228,7 +6228,7 @@
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; }
 then
-  if $READELF -S conftest.so | grep '\.rel\(a\|\)\.dyn' > /dev/null; then
+  if $READELF -S conftest.so | grep '\.\(rela\|rel\)\.dyn' > /dev/null; then
     libc_cv_z_combreloc=yes
   else
     libc_cv_z_combreloc=no
