diff -Nur linux-3.7.3.orig/arch/cris/include/uapi/asm/Kbuild linux-3.7.3/arch/cris/include/uapi/asm/Kbuild
--- linux-3.7.3.orig/arch/cris/include/uapi/asm/Kbuild	2013-01-17 17:47:40.000000000 +0100
+++ linux-3.7.3/arch/cris/include/uapi/asm/Kbuild	2013-01-19 19:14:04.000000000 +0100
@@ -3,3 +3,8 @@
 
 header-y += arch-v10/
 header-y += arch-v32/
+header-y += elf.h
+header-y += page.h
+header-y += ptrace.h
+header-y += user.h
+
diff -Nur linux-3.7.3.orig/arch/cris/include/uapi/asm/arch-v10/Kbuild linux-3.7.3/arch/cris/include/uapi/asm/arch-v10/Kbuild
--- linux-3.7.3.orig/arch/cris/include/uapi/asm/arch-v10/Kbuild	1970-01-01 01:00:00.000000000 +0100
+++ linux-3.7.3/arch/cris/include/uapi/asm/arch-v10/Kbuild	2013-01-17 17:47:40.000000000 +0100
@@ -0,0 +1 @@
+# UAPI Header export list
diff -Nur linux-3.7.3.orig/arch/cris/include/uapi/asm/arch-v32/Kbuild linux-3.7.3/arch/cris/include/uapi/asm/arch-v32/Kbuild
--- linux-3.7.3.orig/arch/cris/include/uapi/asm/arch-v32/Kbuild	1970-01-01 01:00:00.000000000 +0100
+++ linux-3.7.3/arch/cris/include/uapi/asm/arch-v32/Kbuild	2013-01-17 17:47:40.000000000 +0100
@@ -0,0 +1 @@
+# UAPI Header export list
