diff -Nur linux-2.6.32.orig/arch/mips/rb532/devices.c linux-2.6.32/arch/mips/rb532/devices.c
--- linux-2.6.32.orig/arch/mips/rb532/devices.c	2009-12-03 04:51:21.000000000 +0100
+++ linux-2.6.32/arch/mips/rb532/devices.c	2010-02-01 22:29:05.000000000 +0100
@@ -190,8 +190,12 @@
 	}, {
 		.name = "rootfs",
 		.offset = MTDPART_OFS_NXTBLK,
-		.size = MTDPART_SIZ_FULL,
-	}
+		.size = 0x8000000 - 0x400000 - 0x400000,
+	}, {
+		.name = "cfgfs",
+		.offset = 0x8000000 - 0x400000,
+		.size = 0x400000,
+	},
 };
 
 static struct platform_device rb532_led = {
