diff -Nur linux-4.9.20.orig/arch/h8300/Kconfig.cpu linux-4.9.20/arch/h8300/Kconfig.cpu
--- linux-4.9.20.orig/arch/h8300/Kconfig.cpu	2017-03-31 10:32:02.000000000 +0200
+++ linux-4.9.20/arch/h8300/Kconfig.cpu	2017-04-04 08:10:00.132205323 +0200
@@ -96,4 +96,8 @@
         hex "Load offset"
 	default 0
 
+config RAMBASE
+	hex "RAM base address"
+	default 0x400000
+
 endmenu
diff -Nur linux-4.9.60.orig/arch/h8300/boot/dts/h8300h_sim.dts linux-4.9.60/arch/h8300/boot/dts/h8300h_sim.dts
--- linux-4.9.60.orig/arch/h8300/boot/dts/h8300h_sim.dts	2017-11-02 09:49:15.000000000 +0100
+++ linux-4.9.60/arch/h8300/boot/dts/h8300h_sim.dts	2017-12-24 11:30:55.121463793 +0100
@@ -37,7 +37,7 @@
 
 	memory@400000 {
 		device_type = "memory";
-		reg = <0x400000 0x400000>;
+		reg = <0x400000 0x800000>;
 	};
 
 	cpus {
diff -Nur linux-4.9.60.orig/drivers/tty/serial/sh-sci.c linux-4.9.60/drivers/tty/serial/sh-sci.c
--- linux-4.9.60.orig/drivers/tty/serial/sh-sci.c	2017-11-02 09:49:15.000000000 +0100
+++ linux-4.9.60/drivers/tty/serial/sh-sci.c	2017-12-24 11:30:55.121463793 +0100
@@ -986,8 +986,6 @@
 		/* overrun error */
 		if (tty_insert_flip_char(tport, 0, TTY_OVERRUN))
 			copied++;
-
-		dev_notice(port->dev, "overrun error\n");
 	}
 
 	if (status & SCxSR_FER(port)) {
@@ -1060,7 +1058,6 @@
 		tty_insert_flip_char(tport, 0, TTY_OVERRUN);
 		tty_flip_buffer_push(tport);
 
-		dev_dbg(port->dev, "overrun error\n");
 		copied++;
 	}
 
