diff -urN maradns-1.0.26.old/rng/Makefile maradns-1.0.26.dev/rng/Makefile
--- maradns-1.4.05.orig/rng/Makefile	2006-01-02 17:59:33.000000000 +0100
+++ maradns-1.4.05/rng/Makefile	2011-01-21 16:47:29.318658272 +0100
@@ -19,13 +19,13 @@ superclean:
 	rm -f $(ROBJECTS) core make_32bit_tables $(TABLES)
 
 rng-alg-fst.o: rng-alg-fst.c rng-alg-fst.h $(TABLES)
-	$(CC) -c $(FLAGS) -o rng-alg-fst.o rng-alg-fst.c
+	$(CC) -c $(CFLAGS) -o rng-alg-fst.o rng-alg-fst.c
 
 rng-api-fst.o: rng-alg-fst.o rng-api-fst.h 
-	$(CC) -c $(FLAGS) -o rng-api-fst.o rng-api-fst.c
+	$(CC) -c $(CFLAGS) -o rng-api-fst.o rng-api-fst.c
 
 make_32bit_tables: make_32bit_tables.c
-	$(CC) -o make_32bit_tables make_32bit_tables.c
+	$(CC_FOR_BUILD) -o make_32bit_tables make_32bit_tables.c
 
 rng-32bit-tables.h: make_32bit_tables
 	./make_32bit_tables > rng-32bit-tables.h
