diff -Naur linux-4.8.6.orig/include/uapi/linux/atm_zatm.h linux-4.8.6/include/uapi/linux/atm_zatm.h
--- linux-4.8.6.orig/include/uapi/linux/atm_zatm.h	2016-10-31 12:26:46.000000000 +0100
+++ linux-4.8.6/include/uapi/linux/atm_zatm.h	2016-11-07 10:04:37.142753084 +0100
@@ -14,7 +14,6 @@
 
 #include <linux/atmapi.h>
 #include <linux/atmioc.h>
-#include <linux/time.h>
 
 #define ZATM_GETPOOL	_IOW('a',ATMIOC_SARPRV+1,struct atmif_sioc)
 						/* get pool statistics */
