diff -Nur linux-4.8.5.orig/arch/microblaze/kernel/syscall_table.S linux-4.8.5/arch/microblaze/kernel/syscall_table.S
--- linux-4.8.5.orig/arch/microblaze/kernel/syscall_table.S	2016-10-28 09:51:31.000000000 +0200
+++ linux-4.8.5/arch/microblaze/kernel/syscall_table.S	2016-11-01 17:36:41.021129191 +0100
@@ -186,7 +186,7 @@
 	.long sys_getcwd
 	.long sys_capget
 	.long sys_capset		/* 185 */
-	.long sys_ni_syscall		/* sigaltstack */
+	.long sys_sigaltstack
 	.long sys_sendfile
 	.long sys_ni_syscall		/* reserved for streams1 */
 	.long sys_ni_syscall		/* reserved for streams2 */
