diff -Nur gcc-4.9.3.orig/libgcc/config/ia64/fde-glibc.c gcc-4.9.3/libgcc/config/ia64/fde-glibc.c
--- gcc-4.9.3.orig/libgcc/config/ia64/fde-glibc.c	2014-01-02 23:25:22.000000000 +0100
+++ gcc-4.9.3/libgcc/config/ia64/fde-glibc.c	2015-12-01 07:06:43.963906167 +0100
@@ -25,6 +25,8 @@
 /* Locate the FDE entry for a given address, using glibc ld.so routines
    to avoid register/deregister calls at DSO load/unload.  */
 
+#ifndef inhibit_libc
+
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE 1
 #endif
@@ -159,3 +161,4 @@
 
   return data.ret;
 }
+#endif
diff -Nur gcc-4.9.3.orig/libgcc/unwind-generic.h gcc-4.9.3/libgcc/unwind-generic.h
--- gcc-4.9.3.orig/libgcc/unwind-generic.h	2014-01-02 23:25:22.000000000 +0100
+++ gcc-4.9.3/libgcc/unwind-generic.h	2015-12-01 07:00:15.731287428 +0100
@@ -221,7 +221,9 @@
    compatible with the standard ABI for IA-64, we inline these.  */
 
 #ifdef __ia64__
+#ifndef inhibit_libc
 #include <stdlib.h>
+#endif
 
 static inline _Unwind_Ptr
 _Unwind_GetDataRelBase (struct _Unwind_Context *_C)
