diff -Nur gcc-5.3.0.orig/gcc/config/tilegx/linux.h gcc-5.3.0/gcc/config/tilegx/linux.h
--- gcc-5.3.0.orig/gcc/config/tilegx/linux.h	2015-01-05 13:33:28.000000000 +0100
+++ gcc-5.3.0/gcc/config/tilegx/linux.h	2016-04-29 07:14:10.534973350 +0200
@@ -55,8 +55,6 @@
 /* For __clear_cache in libgcc2.c.  */
 #ifdef IN_LIBGCC2
 
-#include <arch/icache.h>
-
 /* Use the minimum page size of 4K.  Alternatively we can call
    getpagesize() but it introduces a libc dependence.  */
 #undef CLEAR_INSN_CACHE
diff -Nur gcc-5.3.0.orig/libgcc/config/tilepro/atomic.h gcc-5.3.0/libgcc/config/tilepro/atomic.h
--- gcc-5.3.0.orig/libgcc/config/tilepro/atomic.h	2015-01-05 13:33:28.000000000 +0100
+++ gcc-5.3.0/libgcc/config/tilepro/atomic.h	2016-04-29 07:15:11.149301459 +0200
@@ -93,7 +93,7 @@
 #endif
 
 #ifdef __tilegx__
-#include <arch/spr_def.h>
+#define SPR_CMPEXCH_VALUE 0x2780 
 #else
 #include <asm/unistd.h>
 #endif
