diff -Naur linux-4.7.3/net/ipv4/netfilter/iptable_filter.c linux-4.7.3.new/net/ipv4/netfilter/iptable_filter.c
--- linux-4.7.3/net/ipv4/netfilter/iptable_filter.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv4/netfilter/iptable_filter.c	2016-09-20 17:34:57.000156904 +0200
@@ -25,7 +25,7 @@
 			    (1 << NF_INET_LOCAL_OUT))
 static int __net_init iptable_filter_table_init(struct net *net);
 
-static const struct xt_table packet_filter = {
+static const struct xt_table __net_initconst packet_filter = {
 	.name		= "filter",
 	.valid_hooks	= FILTER_VALID_HOOKS,
 	.me		= THIS_MODULE,
diff -Naur linux-4.7.3/net/ipv4/netfilter/iptable_mangle.c linux-4.7.3.new/net/ipv4/netfilter/iptable_mangle.c
--- linux-4.7.3/net/ipv4/netfilter/iptable_mangle.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv4/netfilter/iptable_mangle.c	2016-09-20 17:35:04.360257550 +0200
@@ -30,7 +30,7 @@
 
 static int __net_init iptable_mangle_table_init(struct net *net);
 
-static const struct xt_table packet_mangler = {
+static const struct xt_table __net_initconst packet_mangler = {
 	.name		= "mangle",
 	.valid_hooks	= MANGLE_VALID_HOOKS,
 	.me		= THIS_MODULE,
diff -Naur linux-4.7.3/net/ipv4/netfilter/iptable_nat.c linux-4.7.3.new/net/ipv4/netfilter/iptable_nat.c
--- linux-4.7.3/net/ipv4/netfilter/iptable_nat.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv4/netfilter/iptable_nat.c	2016-09-20 17:35:11.847359932 +0200
@@ -20,7 +20,7 @@
 
 static int __net_init iptable_nat_table_init(struct net *net);
 
-static const struct xt_table nf_nat_ipv4_table = {
+static const struct xt_table __net_initconst nf_nat_ipv4_table = {
 	.name		= "nat",
 	.valid_hooks	= (1 << NF_INET_PRE_ROUTING) |
 			  (1 << NF_INET_POST_ROUTING) |
diff -Naur linux-4.7.3/net/ipv4/netfilter/iptable_raw.c linux-4.7.3.new/net/ipv4/netfilter/iptable_raw.c
--- linux-4.7.3/net/ipv4/netfilter/iptable_raw.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv4/netfilter/iptable_raw.c	2016-09-20 17:35:18.424449870 +0200
@@ -12,7 +12,7 @@
 
 static int __net_init iptable_raw_table_init(struct net *net);
 
-static const struct xt_table packet_raw = {
+static const struct xt_table __net_initconst packet_raw = {
 	.name = "raw",
 	.valid_hooks =  RAW_VALID_HOOKS,
 	.me = THIS_MODULE,
diff -Naur linux-4.7.3/net/ipv4/netfilter/iptable_security.c linux-4.7.3.new/net/ipv4/netfilter/iptable_security.c
--- linux-4.7.3/net/ipv4/netfilter/iptable_security.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv4/netfilter/iptable_security.c	2016-09-20 17:35:28.232583992 +0200
@@ -30,7 +30,7 @@
 
 static int __net_init iptable_security_table_init(struct net *net);
 
-static const struct xt_table security_table = {
+static const struct xt_table __net_initconst security_table = {
 	.name		= "security",
 	.valid_hooks	= SECURITY_VALID_HOOKS,
 	.me		= THIS_MODULE,
diff -Naur linux-4.7.3/net/ipv6/netfilter/ip6table_filter.c linux-4.7.3.new/net/ipv6/netfilter/ip6table_filter.c
--- linux-4.7.3/net/ipv6/netfilter/ip6table_filter.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv6/netfilter/ip6table_filter.c	2016-09-20 17:34:01.580392165 +0200
@@ -24,7 +24,7 @@
 
 static int __net_init ip6table_filter_table_init(struct net *net);
 
-static const struct xt_table packet_filter = {
+static const struct xt_table __net_initconst packet_filter = {
 	.name		= "filter",
 	.valid_hooks	= FILTER_VALID_HOOKS,
 	.me		= THIS_MODULE,
diff -Naur linux-4.7.3/net/ipv6/netfilter/ip6table_mangle.c linux-4.7.3.new/net/ipv6/netfilter/ip6table_mangle.c
--- linux-4.7.3/net/ipv6/netfilter/ip6table_mangle.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv6/netfilter/ip6table_mangle.c	2016-09-20 17:34:08.893493248 +0200
@@ -25,7 +25,7 @@
 
 static int __net_init ip6table_mangle_table_init(struct net *net);
 
-static const struct xt_table packet_mangler = {
+static const struct xt_table __net_initconst packet_mangler = {
 	.name		= "mangle",
 	.valid_hooks	= MANGLE_VALID_HOOKS,
 	.me		= THIS_MODULE,
diff -Naur linux-4.7.3/net/ipv6/netfilter/ip6table_nat.c linux-4.7.3.new/net/ipv6/netfilter/ip6table_nat.c
--- linux-4.7.3/net/ipv6/netfilter/ip6table_nat.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv6/netfilter/ip6table_nat.c	2016-09-20 17:34:15.294581725 +0200
@@ -22,7 +22,7 @@
 
 static int __net_init ip6table_nat_table_init(struct net *net);
 
-static const struct xt_table nf_nat_ipv6_table = {
+static const struct xt_table __net_initconst nf_nat_ipv6_table = {
 	.name		= "nat",
 	.valid_hooks	= (1 << NF_INET_PRE_ROUTING) |
 			  (1 << NF_INET_POST_ROUTING) |
diff -Naur linux-4.7.3/net/ipv6/netfilter/ip6table_raw.c linux-4.7.3.new/net/ipv6/netfilter/ip6table_raw.c
--- linux-4.7.3/net/ipv6/netfilter/ip6table_raw.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv6/netfilter/ip6table_raw.c	2016-09-20 17:34:22.939687395 +0200
@@ -11,7 +11,7 @@
 
 static int __net_init ip6table_raw_table_init(struct net *net);
 
-static const struct xt_table packet_raw = {
+static const struct xt_table __net_initconst packet_raw = {
 	.name = "raw",
 	.valid_hooks = RAW_VALID_HOOKS,
 	.me = THIS_MODULE,
diff -Naur linux-4.7.3/net/ipv6/netfilter/ip6table_security.c linux-4.7.3.new/net/ipv6/netfilter/ip6table_security.c
--- linux-4.7.3/net/ipv6/netfilter/ip6table_security.c	2016-09-07 08:35:12.000000000 +0200
+++ linux-4.7.3.new/net/ipv6/netfilter/ip6table_security.c	2016-09-20 17:34:30.348789803 +0200
@@ -29,7 +29,7 @@
 
 static int __net_init ip6table_security_table_init(struct net *net);
 
-static const struct xt_table security_table = {
+static const struct xt_table __net_initconst security_table = {
 	.name		= "security",
 	.valid_hooks	= SECURITY_VALID_HOOKS,
 	.me		= THIS_MODULE,
