diff -Nur linux-3.11.10.orig/arch/microblaze/kernel/setup.c linux-3.11.10/arch/microblaze/kernel/setup.c
--- linux-3.11.10.orig/arch/microblaze/kernel/setup.c	2013-11-29 19:42:37.000000000 +0100
+++ linux-3.11.10/arch/microblaze/kernel/setup.c	2013-12-08 13:02:37.000000000 +0100
@@ -177,7 +177,7 @@
 #else
 	if (!msr) {
 		pr_info("!!!Your kernel not setup MSR instruction but ");
-		pr_cont"CPU have it %x\n", msr);
+		pr_cont("CPU have it %x\n", msr);
 	}
 #endif
 
