diff -Nur binutils-2.23.2.orig/gas/config/tc-arm.c binutils-2.23.2/gas/config/tc-arm.c
--- binutils-2.23.2.orig/gas/config/tc-arm.c	2013-03-25 09:06:21.000000000 +0100
+++ binutils-2.23.2/gas/config/tc-arm.c	2013-09-24 13:26:32.000000000 +0200
@@ -885,6 +885,9 @@
 static inline int
 skip_past_char (char ** str, char c)
 {
+  /* PR gas/14987: Allow for whitespace before the expected character.  */
+  skip_whitespace (*str);
+
   if (**str == c)
     {
       (*str)++;
@@ -5168,6 +5171,9 @@
       return PARSE_OPERAND_SUCCESS;
     }
 
+  /* PR gas/14887: Allow for whitespace after the opening bracket.  */
+  skip_whitespace (p);
+
   if ((reg = arm_reg_parse (&p, REG_TYPE_RN)) == FAIL)
     {
       inst.error = _(reg_expected_msgs[REG_TYPE_RN]);
