diff -Nur linux-3.5.orig//drivers/usb/host/Kconfig linux-3.5/drivers/usb/host/Kconfig
--- linux-3.5.orig//drivers/usb/host/Kconfig	2012-07-21 22:58:29.000000000 +0200
+++ linux-3.5/drivers/usb/host/Kconfig	2012-07-29 11:59:56.000000000 +0200
@@ -66,6 +66,7 @@
 config USB_EHCI_ROOT_HUB_TT
 	bool "Root Hub Transaction Translators"
 	depends on USB_EHCI_HCD || USB_CHIPIDEA_HOST
+	default n
 	---help---
 	  Some EHCI chips have vendor-specific extensions to integrate
 	  transaction translators, so that no OHCI or UHCI companion
@@ -78,7 +79,7 @@
 config USB_EHCI_TT_NEWSCHED
 	bool "Improved Transaction Translator scheduling"
 	depends on USB_EHCI_HCD || USB_CHIPIDEA_HOST
-	default y
+	default n
 	---help---
 	  This changes the periodic scheduling code to fill more of the low
 	  and full speed bandwidth available from the Transaction Translator
