diff -Nur linux-3.16.2.orig/arch/cris/include/arch-v10/arch/Kbuild linux-3.16.2/arch/cris/include/arch-v10/arch/Kbuild
--- linux-3.16.2.orig/arch/cris/include/arch-v10/arch/Kbuild	2014-09-06 01:37:11.000000000 +0200
+++ linux-3.16.2/arch/cris/include/arch-v10/arch/Kbuild	2014-09-26 19:24:50.000000000 +0200
@@ -1 +1,2 @@
 # CRISv10 arch
+header-y += ptrace.h
diff -Nur linux-3.16.2.orig/arch/cris/include/arch-v32/arch/Kbuild linux-3.16.2/arch/cris/include/arch-v32/arch/Kbuild
--- linux-3.16.2.orig/arch/cris/include/arch-v32/arch/Kbuild	2014-09-06 01:37:11.000000000 +0200
+++ linux-3.16.2/arch/cris/include/arch-v32/arch/Kbuild	2014-09-26 19:24:31.000000000 +0200
@@ -1 +1,2 @@
 # CRISv32 arch
+header-y += ptrace.h
