diff -Nur linux-3.19.orig/arch/sparc/boot/Makefile linux-3.19/arch/sparc/boot/Makefile
--- linux-3.19.orig/arch/sparc/boot/Makefile	2015-02-08 20:54:22.000000000 -0600
+++ linux-3.19/arch/sparc/boot/Makefile	2015-02-15 22:58:57.918666227 -0600
@@ -25,9 +25,6 @@
 	$(call if_changed,gzip)
 	@echo '  kernel: $@ is ready'
 
-$(obj)/vmlinux.aout: vmlinux FORCE
-	$(call if_changed,elftoaout)
-	@echo '  kernel: $@ is ready'
 else
 
 $(obj)/zImage: $(obj)/image
@@ -62,6 +59,10 @@
 
 endif
 
+$(obj)/vmlinux.aout: $(obj)/image FORCE
+	$(call if_changed,elftoaout)
+	@echo '  kernel: $@ is ready'
+
 $(obj)/image: vmlinux FORCE
 	$(call if_changed,strip)
 	@echo '  kernel: $@ is ready'
