diff -Nur gcc-4.7.3.orig/gcc/config.host gcc-4.7.3/gcc/config.host
--- gcc-4.7.3.orig/gcc/config.host	Tue Oct 18 08:14:14 2011
+++ gcc-4.7.3/gcc/config.host	Tue Jan  7 04:33:29 2014
@@ -292,6 +292,9 @@
     out_host_hook_obj=host-solaris.o
     host_xmake_file="${host_xmake_file} x-solaris"
     ;;
+  *-*-openbsd*)
+    host_lto_plugin_soname=liblto_plugin.so.0.0
+    ;;
   *-*-linux*)
     out_host_hook_obj=host-linux.o
     host_xmake_file="${host_xmake_file} x-linux"
