diff --no-dereference -urpN a/src/mpfr-longlong.h b/src/mpfr-longlong.h
--- a/src/mpfr-longlong.h	2020-06-18 19:17:18.000000000 +0200
+++ b/src/mpfr-longlong.h	2020-11-25 16:48:03.893628786 +0100
@@ -1283,7 +1283,7 @@ extern UWtype __MPN(udiv_qrnnd) (UWtype
 #endif
 #endif /* __mips */
 
-#if (defined (__mips) && __mips >= 3) && W_TYPE_SIZE == 64
+#if (defined (__mips) && __mips >= 3) && W_TYPE_SIZE == 64 && defined (_LP64)
 #if defined (_MIPS_ARCH_MIPS64R6)
 #define umul_ppmm(w1, w0, u, v) \
   do {									\
