diff -Nur gcc-4.7.3.orig/gcc/config/sparc/linux64.h gcc-4.7.3/gcc/config/sparc/linux64.h
--- gcc-4.7.3.orig/gcc/config/sparc/linux64.h	2011-11-12 02:05:07.000000000 +0100
+++ gcc-4.7.3/gcc/config/sparc/linux64.h	2013-09-19 11:37:50.000000000 +0200
@@ -95,6 +95,9 @@
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
 #define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define MUSL_DYNAMIC_LINKER32 "/lib/ld-musl-sparc.so.1"
+#define MUSL_DYNAMIC_LINKER64 "/lib/ld-musl-sparc.so.1"
+
 
 #ifdef SPARC_BI_ARCH
 
diff -Nur gcc-4.7.3.orig/gcc/config/sparc/linux.h gcc-4.7.3/gcc/config/sparc/linux.h
--- gcc-4.7.3.orig/gcc/config/sparc/linux.h	2011-11-12 02:05:07.000000000 +0100
+++ gcc-4.7.3/gcc/config/sparc/linux.h	2013-09-19 11:37:12.000000000 +0200
@@ -85,6 +85,7 @@
    done.  */
 
 #define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define MUSL_DYNAMIC_LINKER "/lib/ld-musl-sparc.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
