diff -Nur linux-3.14.5.orig/arch/mips/rb532/devices.c linux-3.14.5/arch/mips/rb532/devices.c
--- linux-3.14.5.orig/arch/mips/rb532/devices.c	2014-05-31 21:34:37.000000000 +0200
+++ linux-3.14.5/arch/mips/rb532/devices.c	2014-06-04 17:40:36.318731659 +0200
@@ -191,7 +191,11 @@
 	}, {
 		.name = "rootfs",
 		.offset = MTDPART_OFS_NXTBLK,
-		.size = MTDPART_SIZ_FULL,
+		.size = 108 * 1024 * 1024,
+	}, {
+		.name = "cfgfs",
+		.offset = MTDPART_OFS_NXTBLK,
+		.size = 16 * 1024 * 1024,
 	}
 };
 
