diff -Nur musl-0.9.14.orig/configure musl-0.9.14/configure
--- musl-0.9.14.orig/configure	2013-09-23 23:01:11.000000000 +0200
+++ musl-0.9.14/configure	2013-11-09 20:16:14.000000000 +0100
@@ -225,7 +225,7 @@
 arm*) ARCH=arm ;;
 i?86*) ARCH=i386 ;;
 x86_64*) ARCH=x86_64 ;;
-mips-*|mipsel-*) ARCH=mips ;;
+mips-*|mipsel-*|mips64-*|mips64el-*) ARCH=mips ;;
 microblaze-*) ARCH=microblaze ;;
 powerpc-*) ARCH=powerpc ;;
 unknown) fail "$0: unable to detect target arch; try $0 --target=..." ;;
