diff -Nur glibc-2.22.orig/sysdeps/sparc/sparc32/sem_open.c glibc-2.22/sysdeps/sparc/sparc32/sem_open.c
--- glibc-2.22.orig/sysdeps/sparc/sparc32/sem_open.c	2015-08-05 08:42:21.000000000 +0200
+++ glibc-2.22/sysdeps/sparc/sparc32/sem_open.c	2015-09-04 09:42:55.092504540 +0200
@@ -29,6 +29,7 @@
 #include <sys/mman.h>
 #include <sys/stat.h>
 #include "semaphoreP.h"
+#include <futex-internal.h>
 #include <shm-directory.h>
 
 
