diff -Nur glibc-2.22.orig/sysdeps/unix/sysv/linux/microblaze/sysdep.h glibc-2.22/sysdeps/unix/sysv/linux/microblaze/sysdep.h
--- glibc-2.22.orig/sysdeps/unix/sysv/linux/microblaze/sysdep.h	2015-08-05 08:42:21.000000000 +0200
+++ glibc-2.22/sysdeps/unix/sysv/linux/microblaze/sysdep.h	2015-08-08 10:00:04.148568537 +0200
@@ -16,8 +16,11 @@
    License along with the GNU C Library; if not, see
    <http://www.gnu.org/licenses/>.  */
 
+#ifndef _LINUX_MICROBLAZE_SYSDEP_H
+#define _LINUX_MICROBLAZE_SYSDEP_H 1
+
+#include <sysdeps/unix/sysdep.h>
 #include <sysdeps/microblaze/sysdep.h>
-#include <sys/syscall.h>
 
 /* Defines RTLD_PRIVATE_ERRNO.  */
 #include <dl-sysdep.h>
@@ -305,3 +308,5 @@
 # define PTR_DEMANGLE(var) (void) (var)
 
 #endif /* not __ASSEMBLER__ */
+
+#endif /* _LINUX_MICROBLAZE_SYSDEP_H */
