From 943ef39519a3a4aa514f1fe9431565bf691f01f2 Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@openadk.org>
Date: Mon, 1 Sep 2014 21:02:19 +0200
Subject: [PATCH 1/6] test: disable posix_fallocate64 for small embedded
 systems

Signed-off-by: Waldemar Brodkorb <wbx@openadk.org>
---
 test/unistd/Makefile.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/unistd/Makefile.in b/test/unistd/Makefile.in
index ed33d9a..c884a39 100644
--- a/test/unistd/Makefile.in
+++ b/test/unistd/Makefile.in
@@ -25,6 +25,8 @@ else
 # reordering is not supported, behaves as if POSIXLY_CORRECT would be set
 OPTS_tstgetopt   := -a -b -cfoobar --required foobar --optional=bazbug --none --colou --color --colour random
 endif
+# for embedded systems 4 GB disk space is not available
+TESTS_DISABLED += tst-posix_fallocate64
 
 # getconf.c lives in utils/
 # Testsuite cannot currently be built with O= anyway, so hardcode path here
-- 
1.8.5.2 (Apple Git-48)

