From b46e40edc326638336adc91ccc2aad74c94c764b Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@openadk.org>
Date: Tue, 9 Sep 2014 18:40:22 +0200
Subject: [PATCH 5/6] disable tests for arc as long as binutils is buggy

Signed-off-by: Waldemar Brodkorb <wbx@openadk.org>
---
 test/tls/Makefile.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/tls/Makefile.in b/test/tls/Makefile.in
index 7ab4e2a..0a98337 100644
--- a/test/tls/Makefile.in
+++ b/test/tls/Makefile.in
@@ -12,6 +12,10 @@ ifeq ($(HAVE_SHARED),)
 TESTS_DISABLED := $(TESTS)
 endif
 
+ifeq ($(TARGET_ARCH),arc)
+TESTS_DISABLED += tst-tls12 tst-tls-at-ctor
+endif
+
 # All these tests need tls.h, which is not installed with glibc
 GLIBC_TESTS_DISABLED := $(addsuffix _glibc,$(filter-out $(TESTS_DISABLED),$(TESTS)))
 
-- 
1.8.5.2 (Apple Git-48)

