From cefde9e8caac8ad05409265dcaade63843f2fc71 Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@uclibc-ng.org>
Date: Sun, 24 Aug 2014 21:37:48 +0200
Subject: [PATCH 11/11] get out of the endless while loop, when bind failed

Signed-off-by: Waldemar Brodkorb <wbx@uclibc-ng.org>
---
 test/nptl/tst-cancel4.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/nptl/tst-cancel4.c b/test/nptl/tst-cancel4.c
index 53abf83..4ba4045 100644
--- a/test/nptl/tst-cancel4.c
+++ b/test/nptl/tst-cancel4.c
@@ -1015,6 +1015,8 @@ tf_accept (void *arg)
       if (++tries > 10)
 	{
 	  printf ("%s: too many unsuccessful bind calls\n", __FUNCTION__);
+	  /* prevent endless loop, when bind fails forever */
+	  exit (1);
 	}
 
       strcpy (sun.sun_path, "/tmp/tst-cancel4-socket-1-XXXXXX");
-- 
1.8.5.2 (Apple Git-48)

