diff -Nur uClibc-0.9.33.2.orig/Rules.mak uClibc-0.9.33.2/Rules.mak
--- uClibc-0.9.33.2.orig/Rules.mak	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/Rules.mak	2012-09-21 19:28:45.000000000 +0200
@@ -118,7 +118,7 @@
 LIBC := libc
 SHARED_LIBNAME := $(LIBC).so.$(ABI_VERSION)
 UBACKTRACE_DSO := libubacktrace.so.$(ABI_VERSION)
-ifneq ($(findstring  $(TARGET_ARCH) , hppa64 ia64 mips64 powerpc64 s390x sparc64 x86_64 ),)
+ifneq ($(findstring  $(TARGET_SUBARCH) , hppa64 ia64 mips64 powerpc64 s390x sparc64 x86_64 ),)
 UCLIBC_LDSO_NAME := ld64-uClibc
 ARCH_NATIVE_BIT := 64
 else
diff -Nur uClibc-0.9.33.2.orig/extra/Configs/Config.in uClibc-0.9.33.2/extra/Configs/Config.in
--- uClibc-0.9.33.2.orig/extra/Configs/Config.in	2012-05-15 09:20:09.000000000 +0200
+++ uClibc-0.9.33.2/extra/Configs/Config.in	2012-09-21 19:29:11.000000000 +0200
@@ -235,6 +235,7 @@
 	default "i486" if CONFIG_486
 	default "i586" if CONFIG_586 || CONFIG_586MMX
 	default "i686" if TARGET_ARCH = "i386"
+	default "mips64" if CONFIG_MIPS_ISA_MIPS64
 	default ""
 
 source "extra/Configs/Config.in.arch"
