diff -Nur linux-4.1.20.orig/Makefile linux-4.1.20/Makefile
--- linux-4.1.20.orig/Makefile	2016-03-17 19:11:03.000000000 +0100
+++ linux-4.1.20/Makefile	2016-04-02 13:24:23.000000000 +0200
@@ -359,7 +359,7 @@
 STRIP		= $(CROSS_COMPILE)strip
 OBJCOPY		= $(CROSS_COMPILE)objcopy
 OBJDUMP		= $(CROSS_COMPILE)objdump
-AWK		= awk
+AWK		= gawk
 GENKSYMS	= scripts/genksyms/genksyms
 INSTALLKERNEL  := installkernel
 DEPMOD		= /sbin/depmod
diff -Nur linux-4.1.20.orig/lib/raid6/test/Makefile linux-4.1.20/lib/raid6/test/Makefile
--- linux-4.1.20.orig/lib/raid6/test/Makefile	2016-03-17 19:11:03.000000000 +0100
+++ linux-4.1.20/lib/raid6/test/Makefile	2016-04-02 09:45:15.000000000 +0200
@@ -7,7 +7,7 @@
 OPTFLAGS = -O2			# Adjust as desired
 CFLAGS	 = -I.. -I ../../../include -g $(OPTFLAGS)
 LD	 = ld
-AWK	 = awk -f
+AWK	 = gawk -f
 AR	 = ar
 RANLIB	 = ranlib
 OBJS	 = int1.o int2.o int4.o int8.o int16.o int32.o recov.o algos.o tables.o
