diff -Nur binutils-2.27.orig/bfd/elf32-frv.c binutils-2.27/bfd/elf32-frv.c
--- binutils-2.27.orig/bfd/elf32-frv.c	2016-08-03 09:36:50.000000000 +0200
+++ binutils-2.27/bfd/elf32-frv.c	2016-08-16 14:04:27.941065290 +0200
@@ -5723,10 +5723,6 @@
     }
   if (frvfdpic_got_section (info))
     {
-      BFD_ASSERT (frvfdpic_gotrel_section (info)->size
-		  == (frvfdpic_gotrel_section (info)->reloc_count
-		      * sizeof (Elf32_External_Rel)));
-
       if (frvfdpic_gotfixup_section (info))
 	{
 	  struct elf_link_hash_entry *hgot = elf_hash_table (info)->hgot;
@@ -5766,12 +5762,6 @@
 	    }
 	}
     }
-  if (frvfdpic_pltrel_section (info))
-    {
-      BFD_ASSERT (frvfdpic_pltrel_section (info)->size
-		  == (frvfdpic_pltrel_section (info)->reloc_count
-		      * sizeof (Elf32_External_Rel)));
-    }
 
 
   if (elf_hash_table (info)->dynamic_sections_created)
