From:
https://sourceware.org/bugzilla/show_bug.cgi?id=21017

diff -Nur binutils-2.27.orig/bfd/elf32-microblaze.c binutils-2.27/bfd/elf32-microblaze.c
--- binutils-2.27.orig/bfd/elf32-microblaze.c	2016-08-03 09:36:50.000000000 +0200
+++ binutils-2.27/bfd/elf32-microblaze.c	2017-01-03 04:43:49.173901997 +0100
@@ -2399,6 +2399,7 @@
           tls_type |= (TLS_TLS | TLS_LD);
         dogottls:
           sec->has_tls_reloc = 1;
+	case R_MICROBLAZE_GOTOFF_64:
         case R_MICROBLAZE_GOT_64:
           if (htab->sgot == NULL)
             {
