diff -Nur linux-3.11.5.orig/arch/mips/boot/compressed/decompress.c linux-3.11.5/arch/mips/boot/compressed/decompress.c
--- linux-3.11.5.orig/arch/mips/boot/compressed/decompress.c	2013-10-14 03:14:45.000000000 +0200
+++ linux-3.11.5/arch/mips/boot/compressed/decompress.c	2013-11-08 18:19:47.000000000 +0100
@@ -43,7 +43,7 @@
 /* activate the code for pre-boot environment */
 #define STATIC static
 
-#ifdef CONFIG_KERNEL_GZIP
+#if defined(CONFIG_KERNEL_GZIP) || defined(CONFIG_KERNEL_LZO)
 void *memcpy(void *dest, const void *src, size_t n)
 {
 	int i;
