diff -Nur linux-3.10.37.orig/scripts/Makefile.headersinst linux-3.10.37/scripts/Makefile.headersinst
--- linux-3.10.37.orig/scripts/Makefile.headersinst	2014-04-14 15:42:31.000000000 +0200
+++ linux-3.10.37/scripts/Makefile.headersinst	2014-04-24 17:04:51.980248954 +0200
@@ -107,7 +107,6 @@
 
 targets += $(install-file)
 $(install-file): scripts/headers_install.sh $(input-files1) $(input-files2) $(input-files3) FORCE
-	$(if $(unwanted),$(call cmd,remove),)
 	$(if $(wildcard $(dir $@)),,$(shell mkdir -p $(dir $@)))
 	$(call if_changed,install)
 
