diff -Nur linux-4.1.6.orig/arch/openrisc/kernel/vmlinux.lds.S linux-4.1.6/arch/openrisc/kernel/vmlinux.lds.S
--- linux-4.1.6.orig/arch/openrisc/kernel/vmlinux.lds.S	2015-08-17 05:52:51.000000000 +0200
+++ linux-4.1.6/arch/openrisc/kernel/vmlinux.lds.S	2015-09-01 22:02:13.224692884 +0200
@@ -30,7 +30,7 @@
 #include <asm/cache.h>
 #include <asm-generic/vmlinux.lds.h>
 
-OUTPUT_FORMAT("elf32-or32", "elf32-or32", "elf32-or32")
+OUTPUT_FORMAT("elf32-or1k", "elf32-or1k", "elf32-or1k")
 jiffies = jiffies_64 + 4;
 
 SECTIONS
