--- librpcsecgss-0.19/include/rpcsecgss/rpc/rpc.h~	2013-02-22 08:35:29.949005794 +0100
+++ librpcsecgss-0.19/include/rpcsecgss/rpc/rpc.h	2013-02-22 08:35:31.199005887 +0100
@@ -71,13 +71,6 @@
 
 __BEGIN_DECLS
 
-/* Global variables, protected for multi-threaded applications.  */
-extern fd_set *__rpc_thread_svc_fdset (void) __attribute__ ((__const__));
-#define svc_fdset (*__rpc_thread_svc_fdset ())
-
-extern struct rpc_createerr *__rpc_thread_createerr (void)
-     __attribute__ ((__const__));
-#define get_rpc_createerr() (*__rpc_thread_createerr ())
 /* The people who "engineered" RPC should bee punished for naming the
    data structure and the variable the same.  We cannot always define the
    macro 'rpc_createerr' because this would prevent people from defining
--- librpcsecgss-0.19/include/rpcsecgss/rpc/rpcsecgss_rename.h~	2013-02-22 08:35:34.064007047 +0100
+++ librpcsecgss-0.19/include/rpcsecgss/rpc/rpcsecgss_rename.h	2013-02-22 08:35:35.309007979 +0100
@@ -83,8 +83,6 @@
 #define  rpc_createerr		    rpcsecgss_rpc_createerr
 #define  svc_auth_none		    rpcsecgss_svc_auth_none
 #define  svc_auth_none_ops	    rpcsecgss_svc_auth_none_ops
-#define  __svc_fdset		    rpcsecgss___svc_fdset
-#define  __svc_fdsetsize	    rpcsecgss___svc_fdsetsize
 #define  svc_maxfd		    rpcsecgss_svc_maxfd
 
 #endif	/* _RPCSECGSS_RENAME_H_ */
--- librpcsecgss-0.19/src/clnt_generic.c~	2013-02-22 08:35:43.705020289 +0100
+++ librpcsecgss-0.19/src/clnt_generic.c	2013-02-22 08:35:44.853022749 +0100
@@ -47,8 +47,8 @@ static char *rcsid = "$OpenBSD: clnt_gen
 CLIENT *
 clnt_create(hostname, prog, vers, proto)
 	__const char *hostname;
-	__const u_long prog;
-	__const u_long vers;
+	__const rpcprog_t prog;
+	__const rpcvers_t vers;
 	__const char *proto;
 {
 	struct hostent *h;
--- librpcsecgss-0.19/src/Makefile.am~	2009-05-13 23:33:08.000000000 +0200
+++ librpcsecgss-0.19/src/Makefile.am	2013-02-22 08:38:04.829048549 +0100
@@ -1,7 +1,7 @@
 lib_LTLIBRARIES = librpcsecgss.la
 librpcsecgss_la_CPPFLAGS = $(GSSGLUE_CFLAGS)
 librpcsecgss_la_LDFLAGS = -version-info 3:0:0
-librpcsecgss_la_LIBADD = $(GSSGLUE_LIBS)
+librpcsecgss_la_LIBADD = $(GSSGLUE_LIBS) -ltirpc
 librpcsecgss_la_SOURCES = \
 	auth_gss.c authgss_prot.c svc.c svc_run.c svc_auth.c \
 	svc_auth_none.c svc_auth_unix.c svc_auth_gss.c \
--- librpcsecgss-0.19/src/rpc_commondata.c~	2013-02-22 08:35:39.195012386 +0100
+++ librpcsecgss-0.19/src/rpc_commondata.c	2013-02-22 08:35:40.523014417 +0100
@@ -36,6 +36,4 @@ static char *rcsid = "$OpenBSD: rpc_comm
  * by public interfaces
  */
 struct opaque_auth _null_auth;
-fd_set svc_fdset;
 int svc_maxfd = -1;
-struct rpc_createerr rpc_createerr;

