diff -Nur linux-3.0.4.orig/fs/Kconfig linux-3.0.4/fs/Kconfig
--- linux-3.0.4.orig/fs/Kconfig	2011-08-29 22:56:30.000000000 +0200
+++ linux-3.0.4/fs/Kconfig	2011-10-15 22:08:44.000000000 +0200
@@ -47,7 +47,7 @@
 	def_bool n
 
 config EXPORTFS
-	tristate
+	def_bool y
 
 config FILE_LOCKING
 	bool "Enable POSIX file locking API" if EXPERT
diff -Nur linux-3.0.4.orig/fs/notify/Kconfig linux-3.0.4/fs/notify/Kconfig
--- linux-3.0.4.orig/fs/notify/Kconfig	2011-08-29 22:56:30.000000000 +0200
+++ linux-3.0.4/fs/notify/Kconfig	2011-10-15 22:02:00.000000000 +0200
@@ -1,5 +1,5 @@
 config FSNOTIFY
-	def_bool n
+	def_bool y
 
 source "fs/notify/dnotify/Kconfig"
 source "fs/notify/inotify/Kconfig"
diff -Nur linux-3.11.10.orig/drivers/scsi/Kconfig linux-3.11.10/drivers/scsi/Kconfig
--- linux-3.11.10.orig/drivers/scsi/Kconfig	2013-11-29 19:42:37.000000000 +0100
+++ linux-3.11.10/drivers/scsi/Kconfig	2013-12-27 19:13:21.000000000 +0100
@@ -2,7 +2,7 @@
 
 config SCSI_MOD
        tristate
-       default y if SCSI=n || SCSI=y
+       default y if SCSI=y
        default m if SCSI=m
 
 config RAID_ATTRS
diff -Nur linux-3.11.10.orig/usr/Kconfig linux-3.11.10/usr/Kconfig
--- linux-3.11.10.orig/usr/Kconfig	2013-11-29 19:42:37.000000000 +0100
+++ linux-3.11.10/usr/Kconfig	2013-12-27 19:15:16.000000000 +0100
@@ -47,7 +47,7 @@
 
 config RD_GZIP
 	bool "Support initial ramdisks compressed using gzip" if EXPERT
-	default y
+	default n
 	depends on BLK_DEV_INITRD
 	select DECOMPRESS_GZIP
 	help
