diff -Nur busybox-1.19.4.orig/Config.in busybox-1.19.4/Config.in
--- busybox-1.19.4.orig/Config.in	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4/Config.in	2012-03-20 16:07:34.206229130 +0100
@@ -15,7 +15,7 @@
 
 config DESKTOP
 	bool "Enable options for full-blown desktop systems"
-	default y
+	default n
 	help
 	  Enable options and features which are not essential.
 	  Select this only if you plan to use busybox on full-blown
@@ -32,7 +32,7 @@
 
 config INCLUDE_SUSv2
 	bool "Enable obsolete features removed before SUSv3"
-	default y
+	default n
 	help
 	  This option will enable backwards compatibility with SuSv2,
 	  specifically, old-style numeric options ('command -1 <file>')
diff -Nur busybox-1.19.4.orig/archival/Config.src busybox-1.19.4/archival/Config.src
--- busybox-1.19.4.orig/archival/Config.src	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4/archival/Config.src	2012-03-20 16:07:34.206229130 +0100
@@ -79,7 +79,7 @@
 
 config BUNZIP2
 	bool "bunzip2"
-	default y
+	default n
 	help
 	  bunzip2 is a compression utility using the Burrows-Wheeler block
 	  sorting text compression algorithm, and Huffman coding. Compression
@@ -92,7 +92,7 @@
 
 config BZIP2
 	bool "bzip2"
-	default y
+	default n
 	help
 	  bzip2 is a compression utility using the Burrows-Wheeler block
 	  sorting text compression algorithm, and Huffman coding. Compression
@@ -105,7 +105,7 @@
 
 config CPIO
 	bool "cpio"
-	default y
+	default n
 	help
 	  cpio is an archival utility program used to create, modify, and
 	  extract contents from archives.
@@ -204,19 +204,19 @@
 
 config RPM2CPIO
 	bool "rpm2cpio"
-	default y
+	default n
 	help
 	  Converts a RPM file into a CPIO archive.
 
 config RPM
 	bool "rpm"
-	default y
+	default n
 	help
 	  Mini RPM applet - queries and extracts RPM packages.
 
 config TAR
 	bool "tar"
-	default y
+	default n
 	help
 	  tar is an archiving program. It's commonly used with gzip to
 	  create compressed archives. It's probably the most widely used
@@ -323,7 +323,7 @@
 
 config UNLZMA
 	bool "unlzma"
-	default y
+	default n
 	help
 	  unlzma is a compression utility using the Lempel-Ziv-Markov chain
 	  compression algorithm, and range coding. Compression
@@ -354,7 +354,7 @@
 
 config UNXZ
 	bool "unxz"
-	default y
+	default n
 	help
 	  unxz is a unlzma successor.
 
diff -Nur busybox-1.19.4.orig/editors/Config.src busybox-1.19.4/editors/Config.src
--- busybox-1.19.4.orig/editors/Config.src	2012-02-04 20:24:55.000000000 +0100
+++ busybox-1.19.4/editors/Config.src	2012-03-20 16:07:34.206229130 +0100
@@ -31,7 +31,7 @@
 
 config DIFF
 	bool "diff"
-	default y
+	default n
 	help
 	  diff compares two files or directories and outputs the
 	  differences between them in a form that can be given to
@@ -62,7 +62,7 @@
 
 config SED
 	bool "sed"
-	default y
+	default n
 	help
 	  sed is used to perform text transformations on a file
 	  or input from a pipeline.
diff -Nur busybox-1.19.4.orig/editors/patch.c busybox-1.19.4/editors/patch.c
--- busybox-1.19.4.orig/editors/patch.c	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4/editors/patch.c	2012-03-20 16:07:34.206229130 +0100
@@ -24,7 +24,7 @@
 
 //config:config PATCH
 //config:	bool "patch"
-//config:	default y
+//config:	default n
 //config:	help
 //config:	  Apply a unified diff formatted patch.
 
diff -Nur busybox-1.19.4.orig/findutils/find.c busybox-1.19.4/findutils/find.c
--- busybox-1.19.4.orig/findutils/find.c	2012-02-04 20:24:55.000000000 +0100
+++ busybox-1.19.4/findutils/find.c	2012-03-20 16:07:34.206229130 +0100
@@ -55,7 +55,7 @@
 
 //config:config FIND
 //config:	bool "find"
-//config:	default y
+//config:	default n
 //config:	help
 //config:	  find is used to search your system to find specified files.
 //config:
diff -Nur busybox-1.19.4.orig/findutils/grep.c busybox-1.19.4/findutils/grep.c
--- busybox-1.19.4.orig/findutils/grep.c	2012-02-04 20:24:55.000000000 +0100
+++ busybox-1.19.4/findutils/grep.c	2012-03-20 16:07:34.206229130 +0100
@@ -26,7 +26,7 @@
 
 //config:config GREP
 //config:	bool "grep"
-//config:	default y
+//config:	default n
 //config:	help
 //config:	  grep is used to search files for a specified pattern.
 //config:
diff -Nur busybox-1.19.4.orig/include/applets.src.h busybox-1.19.4/include/applets.src.h
--- busybox-1.19.4.orig/include/applets.src.h	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4/include/applets.src.h	2012-03-20 16:12:28.806228451 +0100
@@ -244,9 +244,9 @@
 IF_MICROCOM(APPLET(microcom, BB_DIR_USR_BIN, BB_SUID_DROP))
 IF_MKDIR(APPLET_NOFORK(mkdir, mkdir, BB_DIR_BIN, BB_SUID_DROP, mkdir))
 IF_MKFS_VFAT(APPLET_ODDNAME(mkdosfs, mkfs_vfat, BB_DIR_SBIN, BB_SUID_DROP, mkfs_vfat))
-IF_MKFS_EXT2(APPLET_ODDNAME(mke2fs, mkfs_ext2, BB_DIR_SBIN, BB_SUID_DROP, mkfs_ext2))
+//IF_MKFS_EXT2(APPLET_ODDNAME(mke2fs, mkfs_ext2, BB_DIR_SBIN, BB_SUID_DROP, mkfs_ext2))
 IF_MKFIFO(APPLET_NOEXEC(mkfifo, mkfifo, BB_DIR_USR_BIN, BB_SUID_DROP, mkfifo))
-IF_MKFS_EXT2(APPLET_ODDNAME(mkfs.ext2, mkfs_ext2, BB_DIR_SBIN, BB_SUID_DROP, mkfs_ext2))
+//IF_MKFS_EXT2(APPLET_ODDNAME(mkfs.ext2, mkfs_ext2, BB_DIR_SBIN, BB_SUID_DROP, mkfs_ext2))
 //IF_MKE2FS(APPLET_ODDNAME(mkfs.ext3, mke2fs, BB_DIR_SBIN, BB_SUID_DROP, mkfs_ext3))
 IF_MKFS_MINIX(APPLET_ODDNAME(mkfs.minix, mkfs_minix, BB_DIR_SBIN, BB_SUID_DROP, mkfs_minix))
 IF_MKFS_REISER(APPLET_ODDNAME(mkfs.reiser, mkfs_reiser, BB_DIR_SBIN, BB_SUID_DROP, mkfs_reiser))
diff -Nur busybox-1.19.4.orig/miscutils/Config.src busybox-1.19.4/miscutils/Config.src
--- busybox-1.19.4.orig/miscutils/Config.src	2012-02-04 20:24:55.000000000 +0100
+++ busybox-1.19.4/miscutils/Config.src	2012-03-20 16:07:34.206229130 +0100
@@ -362,7 +362,7 @@
 
 config HDPARM
 	bool "hdparm"
-	default y
+	default n
 	select PLATFORM_LINUX
 	help
 	  Get/Set hard drive parameters. Primarily intended for ATA
@@ -600,7 +600,7 @@
 
 config WATCHDOG
 	bool "watchdog"
-	default y
+	default n
 	select PLATFORM_LINUX
 	help
 	  The watchdog utility is used with hardware or software watchdog
diff -Nur busybox-1.19.4.orig/networking/Config.src busybox-1.19.4/networking/Config.src
--- busybox-1.19.4.orig/networking/Config.src	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4/networking/Config.src	2012-03-20 16:07:34.206229130 +0100
@@ -64,7 +64,7 @@
 
 config BRCTL
 	bool "brctl"
-	default y
+	default n
 	select PLATFORM_LINUX
 	help
 	  Manage ethernet bridges.
@@ -97,7 +97,7 @@
 
 config ETHER_WAKE
 	bool "ether-wake"
-	default y
+	default n
 	select PLATFORM_LINUX
 	help
 	  Send a magic packet to wake up sleeping machines.
@@ -498,7 +498,7 @@
 
 config IP
 	bool "ip"
-	default y
+	default n
 	select PLATFORM_LINUX
 	help
 	  The "ip" applet is a TCP/IP interface configuration and routing
@@ -935,7 +935,7 @@
 
 config WGET
 	bool "wget"
-	default y
+	default n
 	help
 	  wget is a utility for non-interactive download of files from HTTP,
 	  HTTPS, and FTP servers.
diff -Nur busybox-1.19.4.orig/util-linux/Config.src busybox-1.19.4/util-linux/Config.src
--- busybox-1.19.4.orig/util-linux/Config.src	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4/util-linux/Config.src	2012-03-20 16:09:26.662228452 +0100
@@ -246,13 +246,6 @@
 	  check for and attempt to repair any corruption that occurs to a minix
 	  filesystem.
 
-config MKFS_EXT2
-	bool "mkfs_ext2"
-	default y
-	select PLATFORM_LINUX
-	help
-	  Utility to create EXT2 filesystems.
-
 config MKFS_MINIX
 	bool "mkfs_minix"
 	default y
@@ -386,7 +379,7 @@
 
 config LSPCI
 	bool "lspci"
-	default y
+	default n
 	#select PLATFORM_LINUX
 	help
 	  lspci is a utility for displaying information about PCI buses in the
@@ -396,7 +389,7 @@
 
 config LSUSB
 	bool "lsusb"
-	default y
+	default n
 	#select PLATFORM_LINUX
 	help
 	  lsusb is a utility for displaying information about USB buses in the
diff -Nur busybox-1.19.4.orig/util-linux/Kbuild.src busybox-1.19.4/util-linux/Kbuild.src
--- busybox-1.19.4.orig/util-linux/Kbuild.src	2012-02-04 20:34:24.000000000 +0100
+++ busybox-1.19.4/util-linux/Kbuild.src	2012-03-20 16:10:21.682228369 +0100
@@ -27,7 +27,6 @@
 lib-$(CONFIG_LSPCI)             += lspci.o
 lib-$(CONFIG_LSUSB)             += lsusb.o
 lib-$(CONFIG_MDEV)              += mdev.o
-lib-$(CONFIG_MKFS_EXT2)         += mkfs_ext2.o
 lib-$(CONFIG_MKFS_MINIX)        += mkfs_minix.o
 lib-$(CONFIG_MKFS_REISER)       += mkfs_reiser.o
 lib-$(CONFIG_MKFS_VFAT)         += mkfs_vfat.o
