diff -Nur busybox-1.17.0.orig/archival/Config.src busybox-1.17.0/archival/Config.src
--- busybox-1.17.0.orig/archival/Config.src	2010-06-25 01:32:35.000000000 +0200
+++ busybox-1.17.0/archival/Config.src	2010-07-16 02:06:58.336016174 +0200
@@ -39,7 +39,7 @@
 
 config AR
 	bool "ar"
-	default y
+	default n
 	help
 	  ar is an archival utility program used to create, modify, and
 	  extract contents from archives. An archive is a single file holding
@@ -79,7 +79,7 @@
 
 config BUNZIP2
 	bool "bunzip2"
-	default y
+	default n
 	help
 	  bunzip2 is a compression utility using the Burrows-Wheeler block
 	  sorting text compression algorithm, and Huffman coding. Compression
@@ -92,7 +92,7 @@
 
 config BZIP2
 	bool "bzip2"
-	default y
+	default n
 	help
 	  bzip2 is a compression utility using the Burrows-Wheeler block
 	  sorting text compression algorithm, and Huffman coding. Compression
@@ -105,7 +105,7 @@
 
 config CPIO
 	bool "cpio"
-	default y
+	default n
 	help
 	  cpio is an archival utility program used to create, modify, and
 	  extract contents from archives.
@@ -204,19 +204,19 @@
 
 config RPM2CPIO
 	bool "rpm2cpio"
-	default y
+	default n
 	help
 	  Converts an RPM file into a CPIO archive.
 
 config RPM
 	bool "rpm"
-	default y
+	default n
 	help
 	  Mini RPM applet - queries and extracts RPM packages.
 
 config TAR
 	bool "tar"
-	default y
+	default n
 	help
 	  tar is an archiving program. It's commonly used with gzip to
 	  create compressed archives. It's probably the most widely used
@@ -323,7 +323,7 @@
 
 config UNLZMA
 	bool "unlzma"
-	default y
+	default n
 	help
 	  unlzma is a compression utility using the Lempel-Ziv-Markov chain
 	  compression algorithm, and range coding. Compression
@@ -354,7 +354,7 @@
 
 config UNXZ
 	bool "unxz"
-	default y
+	default n
 	help
 	  unxz is a unlzma successor.
 
diff -Nur busybox-1.17.0.orig/editors/Config.src busybox-1.17.0/editors/Config.src
--- busybox-1.17.0.orig/editors/Config.src	2010-06-24 04:40:43.000000000 +0200
+++ busybox-1.17.0/editors/Config.src	2010-07-16 02:05:47.485881936 +0200
@@ -31,7 +31,7 @@
 
 config DIFF
 	bool "diff"
-	default y
+	default n
 	help
 	  diff compares two files or directories and outputs the
 	  differences between them in a form that can be given to
@@ -62,13 +62,13 @@
 
 config PATCH
 	bool "patch"
-	default y
+	default n
 	help
 	  Apply a unified diff formatted patch.
 
 config SED
 	bool "sed"
-	default y
+	default n
 	help
 	  sed is used to perform text transformations on a file
 	  or input from a pipeline.
diff -Nur busybox-1.17.0.orig/networking/Config.src busybox-1.17.0/networking/Config.src
--- busybox-1.17.0.orig/networking/Config.src	2010-06-27 04:21:56.000000000 +0200
+++ busybox-1.17.0/networking/Config.src	2010-07-16 02:05:16.665861002 +0200
@@ -961,7 +961,7 @@
 
 config WGET
 	bool "wget"
-	default y
+	default n
 	help
 	  wget is a utility for non-interactive download of files from HTTP,
 	  HTTPS, and FTP servers.
