diff -Nur busybox-1.18.1.orig/Makefile.flags busybox-1.18.1/Makefile.flags
--- busybox-1.18.1.orig/Makefile.flags	2010-12-20 01:41:26.000000000 +0100
+++ busybox-1.18.1/Makefile.flags	2011-01-27 21:24:51.000000000 +0100
@@ -57,18 +57,6 @@
 # be fixed..
 #CFLAGS += $(call cc-option,-Wconversion,)
 
-ifneq ($(CONFIG_DEBUG),y)
-CFLAGS += $(call cc-option,-Os,)
-else
-CFLAGS += $(call cc-option,-g,)
-#CFLAGS += "-D_FORTIFY_SOURCE=2"
-ifeq ($(CONFIG_DEBUG_PESSIMIZE),y)
-CFLAGS += $(call cc-option,-O0,)
-else
-CFLAGS += $(call cc-option,-Os,)
-endif
-endif
-
 # If arch/$(ARCH)/Makefile did not override it (with, say, -fPIC)...
 ARCH_FPIC ?= -fpic
 ARCH_FPIE ?= -fpie
