diff -Nur busybox-1.17.1.orig/include/libbb.h busybox-1.17.1/include/libbb.h
--- busybox-1.17.1.orig/include/libbb.h	Sun Jul 25 00:12:43 2010
+++ busybox-1.17.1/include/libbb.h	Thu Jul 29 20:27:07 2010
@@ -40,6 +40,7 @@
 /* Try to pull in PATH_MAX */
 #include <limits.h>
 #include <sys/param.h>
+#if !(defined __APPLE__ || defined BSD )
 #ifdef HAVE_MNTENT_H
 #include <mntent.h>
 #endif
@@ -51,6 +52,7 @@
 #include <selinux/context.h>
 #include <selinux/flask.h>
 #include <selinux/av_permissions.h>
+#endif
 #endif
 #if ENABLE_LOCALE_SUPPORT
 # include <locale.h>
diff -Nur busybox-1.17.1.orig/include/platform.h busybox-1.17.1/include/platform.h
--- busybox-1.17.1.orig/include/platform.h	Tue Jul  6 04:25:54 2010
+++ busybox-1.17.1/include/platform.h	Thu Jul 29 20:28:50 2010
@@ -150,18 +150,19 @@
 
 /* ---- Endian Detection ------------------------------------ */
 
+# include <sys/param.h>
 #if defined(__digital__) && defined(__unix__)
 # include <sex.h>
 # define __BIG_ENDIAN__ (BYTE_ORDER == BIG_ENDIAN)
 # define __BYTE_ORDER BYTE_ORDER
-#elif defined __FreeBSD__
+#elif defined __FreeBSD__ || defined __APPLE__
 # include <sys/resource.h>	/* rlimit */
 # include <machine/endian.h>
 # define bswap_64 __bswap64
 # define bswap_32 __bswap32
 # define bswap_16 __bswap16
 # define __BIG_ENDIAN__ (_BYTE_ORDER == _BIG_ENDIAN)
-#elif !defined __APPLE__
+#elif !defined BSD
 # include <byteswap.h>
 # include <endian.h>
 #endif
@@ -172,9 +173,15 @@
 #elif defined(__BYTE_ORDER) && __BYTE_ORDER == __BIG_ENDIAN
 # define BB_BIG_ENDIAN 1
 # define BB_LITTLE_ENDIAN 0
+#elif defined(_BYTE_ORDER) && _BYTE_ORDER == _BIG_ENDIAN
+# define BB_BIG_ENDIAN 1
+# define BB_LITTLE_ENDIAN 0
 #elif (defined(__BYTE_ORDER) && __BYTE_ORDER == __LITTLE_ENDIAN) || defined(__386__)
 # define BB_BIG_ENDIAN 0
 # define BB_LITTLE_ENDIAN 1
+#elif (defined(_BYTE_ORDER) && _BYTE_ORDER == _LITTLE_ENDIAN) || defined(__386__)
+# define BB_BIG_ENDIAN 0
+# define BB_LITTLE_ENDIAN 1
 #else
 # error "Can't determine endianness"
 #endif
@@ -230,12 +237,19 @@
 /* ---- Compiler dependent settings ------------------------- */
 
 #if (defined __digital__ && defined __unix__) \
- || defined __APPLE__ || defined __FreeBSD__
+ || defined __APPLE__ || defined __FreeBSD__ || defined BSD
 # undef HAVE_MNTENT_H
 # undef HAVE_SYS_STATFS_H
 #else
 # define HAVE_MNTENT_H 1
 # define HAVE_SYS_STATFS_H 1
+#endif
+
+/* ---- Networking ------------------------------------------ */
+
+#if defined __OpenBSD__
+#include <netinet/in.h>
+#include <arpa/inet.h>
 #endif
 
 /*----- Kernel versioning ------------------------------------*/
diff -Nur busybox-1.17.1.orig/scripts/gen_build_files.sh busybox-1.17.1/scripts/gen_build_files.sh
--- busybox-1.17.1.orig/scripts/gen_build_files.sh	Sun Jul 25 00:12:56 2010
+++ busybox-1.17.1/scripts/gen_build_files.sh	Thu Jul 29 20:27:07 2010
@@ -48,7 +48,7 @@
 fi
 
 # (Re)generate */Kbuild and */Config.in
-{ cd -- "$srctree" && find -type d; } | while read -r d; do
+{ cd -- "$srctree" && find . -type d; } | while read -r d; do
 	d="${d#./}"
 
 	src="$srctree/$d/Kbuild.src"
