diff -Nur busybox-1.26.0.orig/modutils/modutils.c busybox-1.26.0/modutils/modutils.c
--- busybox-1.26.0.orig/modutils/modutils.c	2016-10-03 23:16:44.000000000 +0200
+++ busybox-1.26.0/modutils/modutils.c	2016-12-23 06:15:16.694563841 +0100
@@ -7,17 +7,12 @@
  */
 #include "modutils.h"
 
-#ifdef __UCLIBC__
-extern int init_module(void *module, unsigned long len, const char *options);
-extern int delete_module(const char *module, unsigned int flags);
-#else
 # include <sys/syscall.h>
 # define init_module(mod, len, opts) syscall(__NR_init_module, mod, len, opts)
 # if defined(__NR_finit_module)
 #  define finit_module(fd, uargs, flags) syscall(__NR_finit_module, fd, uargs, flags)
 # endif
 # define delete_module(mod, flags) syscall(__NR_delete_module, mod, flags)
-#endif
 
 static module_entry *helper_get_module(module_db *db, const char *module, int create)
 {
