diff -Nur busybox-1.20.2.orig/Makefile busybox-1.20.2/Makefile
--- busybox-1.20.2.orig/Makefile	2012-07-02 16:20:27.000000000 +0200
+++ busybox-1.20.2/Makefile	2013-08-12 13:36:00.000000000 +0200
@@ -318,10 +318,9 @@
 CFLAGS		:= $(CFLAGS)
 # Added only to final link stage of busybox binary
 CFLAGS_busybox	:= $(CFLAGS_busybox)
-CPPFLAGS	:= $(CPPFLAGS)
+CPPFLAGS	:= $(CPPFLAGS) $(EXTRA_CPPFLAGS)
 AFLAGS		:= $(AFLAGS)
-LDFLAGS		:= $(LDFLAGS)
-LDLIBS		:=
+LDFLAGS		:= $(LDFLAGS) $(EXTRA_LDFLAGS)
 
 # Read KERNELRELEASE from .kernelrelease (if it exists)
 KERNELRELEASE = $(shell cat .kernelrelease 2> /dev/null)
