diff -Nur busybox-1.26.0.orig/networking/Config.src busybox-1.26.0/networking/Config.src
--- busybox-1.26.0.orig/networking/Config.src	2016-12-12 08:46:14.000000000 +0100
+++ busybox-1.26.0/networking/Config.src	2016-12-25 09:39:42.715064866 +0100
@@ -53,7 +53,7 @@
 config IFUPDOWN_UDHCPC_CMD_OPTIONS
 	string "ifup udhcpc command line options"
 	default "-R -n"
-	depends on IFUPDOWN && UDHCPC
+	depends on IFUP && UDHCPC
 	help
 	  Command line options to pass to udhcpc from ifup.
 	  Intended to alter options not available in /etc/network/interfaces.
