diff -Nur busybox-1.20.2.orig/scripts/gen_build_files.sh busybox-1.20.2/scripts/gen_build_files.sh
--- busybox-1.20.2.orig/scripts/gen_build_files.sh	2012-06-26 13:35:45.000000000 +0000
+++ busybox-1.20.2/scripts/gen_build_files.sh	2013-04-29 12:18:26.089999426 +0000
@@ -61,7 +61,7 @@
 
 # (Re)generate */Kbuild and */Config.in
 # We skip .dotdirs - makes git/svn/etc users happier
-{ cd -- "$srctree" && find . -type d -not '(' -name '.?*' -prune ')'; } \
+{ cd -- "$srctree" && find . -type d ! '(' -name '.?*' -prune ')'; } \
 | while read -r d; do
 	d="${d#./}"
 
