diff -Nur busybox-1.20.2.orig/Makefile.flags busybox-1.20.2/Makefile.flags
--- busybox-1.20.2.orig/Makefile.flags	2012-06-26 15:35:45.000000000 +0200
+++ busybox-1.20.2/Makefile.flags	2013-08-13 14:25:53.000000000 +0200
@@ -50,7 +50,7 @@
 # -fno-guess-branch-probability: prohibit pseudo-random guessing
 # of branch probabilities (hopefully makes bloatcheck more stable):
 CFLAGS += $(call cc-option,-fno-guess-branch-probability,)
-CFLAGS += $(call cc-option,-funsigned-char -static-libgcc,)
+CFLAGS += $(call cc-option,-funsigned-char,)
 CFLAGS += $(call cc-option,-falign-functions=1 -falign-jumps=1 -falign-labels=1 -falign-loops=1,)
 
 # FIXME: These warnings are at least partially to be concerned about and should
