diff -Nur busybox-1.21.1.orig/include/libbb.h busybox-1.21.1/include/libbb.h
--- busybox-1.21.1.orig/include/libbb.h	2013-06-29 16:58:06.000000000 +0200
+++ busybox-1.21.1/include/libbb.h	2013-11-02 16:55:04.000000000 +0100
@@ -37,7 +37,7 @@
 #include <libgen.h> /* dirname,basename */
 #undef basename
 #define basename dont_use_basename
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 #include <sys/socket.h>
