diff -Nur busybox-1.21.1.orig/scripts/basic/docproc.c busybox-1.21.1/scripts/basic/docproc.c
--- busybox-1.21.1.orig/scripts/basic/docproc.c	Sat May 11 17:30:44 2013
+++ busybox-1.21.1/scripts/basic/docproc.c	Thu Jan 16 03:00:39 2014
@@ -39,7 +39,6 @@
 #include <limits.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#include <alloca.h>
 
 /* exitstatus is used to keep track of any failing calls to kernel-doc,
  * but execution continues. */
diff -Nur busybox-1.21.1.orig/scripts/basic/fixdep.c busybox-1.21.1/scripts/basic/fixdep.c
--- busybox-1.21.1.orig/scripts/basic/fixdep.c	Sat May 11 17:30:44 2013
+++ busybox-1.21.1/scripts/basic/fixdep.c	Thu Jan 16 03:00:33 2014
@@ -113,7 +113,6 @@
 #include <limits.h>
 #include <ctype.h>
 #include <arpa/inet.h>
-#include <alloca.h>
 
 /* bbox: not needed
 #define INT_CONF ntohl(0x434f4e46)
