diff -Nur gcc-4.5.3.orig/gcc/configure gcc-4.5.3/gcc/configure
--- gcc-4.5.3.orig/gcc/configure	2010-12-03 13:35:37.000000000 +0100
+++ gcc-4.5.3/gcc/configure	2011-10-19 11:29:26.155085565 +0200
@@ -24579,16 +24579,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $gcc_cv_as_gnu_unique_object" >&5
 $as_echo "$gcc_cv_as_gnu_unique_object" >&6; }
 if test $gcc_cv_as_gnu_unique_object = yes; then
-  # Also check for ld.so support, i.e. glibc 2.11 or higher.
-   if test x$host = x$build -a x$host = x$target &&
-       glibcver=`ldd --version 2>/dev/null | sed 's/.* //;q'`; then
-      glibcmajor=`expr "$glibcver" : "\([0-9]*\)"`
-      glibcminor=`expr "$glibcver" : "[2-9]*\.\([0-9]*\)"`
-      glibcnum=`expr $glibcmajor \* 1000 + $glibcminor`
-      if test "$glibcnum" -ge 2011 ; then
         enable_gnu_unique_object=yes
-      fi
-    fi
 fi
 fi
 
