diff -Nur linux-4.1.10.orig/drivers/media/v4l2-core/videobuf2-core.c linux-4.1.10/drivers/media/v4l2-core/videobuf2-core.c
--- linux-4.1.10.orig/drivers/media/v4l2-core/videobuf2-core.c	2015-10-03 13:49:38.000000000 +0200
+++ linux-4.1.10/drivers/media/v4l2-core/videobuf2-core.c	2015-10-18 18:18:47.000000000 +0200
@@ -1245,7 +1245,6 @@
 		return;
 
 	__check_once = true;
-	__WARN();
 
 	pr_warn_once("use of bytesused == 0 is deprecated and will be removed in the future,\n");
 	if (vb->vb2_queue->allow_zero_bytesused)
