diff -Nur newlib-xtensa.orig/configure newlib-xtensa/configure
--- newlib-xtensa.orig/configure	2017-12-29 15:26:12.000000000 +0100
+++ newlib-xtensa/configure	2017-12-29 15:35:01.420208845 +0100
@@ -6847,7 +6847,7 @@
 
 # Some systems (e.g., one of the i386-aix systems the gas testers are
 # using) don't handle "\$" correctly, so don't use it here.
-tooldir='${exec_prefix}'/${target_noncanonical}
+tooldir='${exec_prefix}'
 build_tooldir=${tooldir}
 
 # Create a .gdbinit file which runs the one in srcdir
