diff -Nur linux-3.13.3.orig/arch/x86/boot/compressed/mkpiggy.c linux-3.13.3/arch/x86/boot/compressed/mkpiggy.c
--- linux-3.13.3.orig/arch/x86/boot/compressed/mkpiggy.c	2014-02-13 23:00:14.000000000 +0100
+++ linux-3.13.3/arch/x86/boot/compressed/mkpiggy.c	2014-02-17 11:09:06.000000000 +0100
@@ -29,7 +29,14 @@
 #include <stdio.h>
 #include <string.h>
 #include <inttypes.h>
-#include <tools/le_byteshift.h>
+
+static uint32_t getle32(const void *p)
+{
+	const uint8_t *cp = p;
+
+	return (uint32_t)cp[0] + ((uint32_t)cp[1] << 8) +
+		((uint32_t)cp[2] << 16) + ((uint32_t)cp[3] << 24);
+}
 
 int main(int argc, char *argv[])
 {
@@ -63,7 +70,7 @@
 	}
 
 	ilen = ftell(f);
-	olen = get_unaligned_le32(&olen);
+	olen = getle32(&olen);
 
 	/*
 	 * Now we have the input (compressed) and output (uncompressed)
