diff -Nur linux-3.4.2.orig/scripts/headers_check.pl linux-3.4.2/scripts/headers_check.pl
--- linux-3.4.2.orig/scripts/headers_check.pl	2012-06-09 17:36:33.000000000 +0200
+++ linux-3.4.2/scripts/headers_check.pl	2012-06-15 12:45:43.000000000 +0200
@@ -18,7 +18,7 @@
 #
 # 3) Check for leaked CONFIG_ symbols
 
-use strict;
+#use strict;
 use File::Basename;
 
 my ($dir, $arch, @files) = @ARGV;
diff -Nur linux-3.4.2.orig/scripts/headers_install.pl linux-3.4.2/scripts/headers_install.pl
--- linux-3.4.2.orig/scripts/headers_install.pl	2012-06-09 17:36:33.000000000 +0200
+++ linux-3.4.2/scripts/headers_install.pl	2012-06-15 12:45:11.000000000 +0200
@@ -16,7 +16,7 @@
 # 2) Drop include of compiler.h
 # 3) Drop all sections defined out by __KERNEL__ (using unifdef)
 
-use strict;
+#use strict;
 
 my ($readdir, $installdir, $arch, @files) = @ARGV;
 
