diff -Nur linux-3.5.2.orig/arch/mips/Kconfig linux-3.5.2/arch/mips/Kconfig
--- linux-3.5.2.orig/arch/mips/Kconfig	2012-08-15 16:55:25.000000000 +0200
+++ linux-3.5.2/arch/mips/Kconfig	2012-08-23 11:10:29.000000000 +0200
@@ -31,7 +31,6 @@
 	select HAVE_MEMBLOCK_NODE_MAP
 	select ARCH_DISCARD_MEMBLOCK
 	select GENERIC_SMP_IDLE_THREAD
-	select BUILDTIME_EXTABLE_SORT
 	select GENERIC_CLOCKEVENTS
 	select GENERIC_CMOS_UPDATE
 
diff -Nur linux-3.5.2.orig/arch/x86/Kconfig linux-3.5.2/arch/x86/Kconfig
--- linux-3.5.2.orig/arch/x86/Kconfig	2012-08-15 16:55:25.000000000 +0200
+++ linux-3.5.2/arch/x86/Kconfig	2012-08-23 11:10:17.000000000 +0200
@@ -85,7 +85,6 @@
 	select DCACHE_WORD_ACCESS
 	select GENERIC_SMP_IDLE_THREAD
 	select HAVE_ARCH_SECCOMP_FILTER
-	select BUILDTIME_EXTABLE_SORT
 	select GENERIC_CMOS_UPDATE
 	select CLOCKSOURCE_WATCHDOG
 	select GENERIC_CLOCKEVENTS
