diff -urN libart.old/Makefile.in libart.dev/Makefile.in
--- libart_lgpl-2.3.17.orig/Makefile.in	2005-01-21 10:52:11.000000000 +0100
+++ libart_lgpl-2.3.17/Makefile.in	2011-02-26 22:48:38.000000000 +0100
@@ -440,9 +440,9 @@ testuta$(EXEEXT): $(testuta_OBJECTS) $(t
 	@rm -f testuta$(EXEEXT)
 	$(LINK) $(testuta_LDFLAGS) $(testuta_OBJECTS) $(testuta_LDADD) $(LIBS)
 
-gen_art_config$(EXEEXT): $(gen_art_config_OBJECTS) $(gen_art_config_DEPENDENCIES)
+gen_art_config$(EXEEXT):
 	@rm -f gen_art_config$(EXEEXT)
-	$(LINK) $(gen_art_config_LDFLAGS) $(gen_art_config_OBJECTS) $(gen_art_config_LDADD) $(LIBS)
+	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) -I. -DLIBART_COMPILATION -DHAVE_CONFIG_H -o $@ gen_art_config.c
 
 install-binSCRIPTS: $(bin_SCRIPTS)
 	@$(NORMAL_INSTALL)
