--- w-openjdk-6-1.orig/openjdk-6.orig/jdk/make/common/shared/Sanity.gmk	2011-02-28 17:06:10.000000000 +0100
+++ w-openjdk-6-1/openjdk-6/jdk/make/common/shared/Sanity.gmk	2011-04-05 14:05:29.691807661 +0200
@@ -793,7 +793,7 @@
 # Check for availability of FreeType (OpenJDK specific)
 ######################################################
 
-ifdef OPENJDK
+ifdef OPENJDK_DISABLED
 
 #name of test program that prints out "Failed" if freetype is not good enough
 FREETYPE_VERSION_CHECK_NAME = freetype_versioncheck
