
 Fails when cross-compilation is done, disable it.

--- w-openjdk-6-1.orig/openjdk-6/hotspot/make/linux/makefiles/buildtree.make	2011-02-28 17:03:13.000000000 +0100
+++ w-openjdk-6-1/openjdk-6/hotspot/make/linux/makefiles/buildtree.make	2011-04-11 12:47:31.000000000 +0200
@@ -352,12 +352,8 @@
 	echo "if [ -z \$$JAVA_HOME ]; then { $(NO_JAVA_HOME_MSG); exit 0; }; fi"; \
 	echo "if ! \$${JAVA_HOME}/bin/java $(JAVA_FLAG) -fullversion 2>&1 > /dev/null"; \
 	echo "then"; \
-	echo "  $(WRONG_DATA_MODE_MSG); exit 0;"; \
+	echo "  exit 0;"; \
 	echo "fi"; \
-	echo "rm -f Queens.class"; \
-	echo "\$${JAVA_HOME}/bin/javac -d . $(GAMMADIR)/make/test/Queens.java"; \
-	echo '[ -f gamma_g ] && { gamma=gamma_g; }'; \
-	echo './$${gamma:-gamma} $(TESTFLAGS) Queens < /dev/null'; \
 	) > $@
 	$(QUIETLY) chmod +x $@
 
