diff -Nur w-openjdk-6.orig//openjdk-6/jdk/make/sun/xawt/Makefile w-openjdk-6/openjdk-6/jdk/make/sun/xawt/Makefile
--- w-openjdk-6.orig//openjdk-6/jdk/make/sun/xawt/Makefile	2011-02-28 17:06:12.000000000 +0100
+++ w-openjdk-6/openjdk-6/jdk/make/sun/xawt/Makefile	2011-04-11 19:50:34.000000000 +0200
@@ -237,7 +237,9 @@
 	$(BOOT_JAVA_CMD) -cp $(WRAPPER_GENERATOR_TEMPDIR) WrapperGenerator \
 	    $(SIZER_DIR) $(XLIBTYPES) "sizer" $(subst .,,$(suffix $(basename $@)))
 
-$(SIZES): $(SIZERS)
+#$(SIZES): $(SIZERS)
+$(SIZES):
+	@$(prep-target)
 	@if [ "$(DOHACK)$@" = "true$(PREDEFINED_SIZES)" ]; then \
 	    $(ECHO) COPYING $(PREDEFINED_SIZES_TMPL) into $@; \
 	    $(CP) $(PREDEFINED_SIZES_TMPL) $@; \
