--- w-openjdk-6-1.orig/icedtea6-1.10/Makefile.in	2011-03-02 20:48:45.000000000 +0100
+++ w-openjdk-6-1/icedtea6-1.10/Makefile.in	2011-04-02 23:33:17.000000000 +0200
@@ -1576,7 +1576,7 @@ clean-overlay:
 # FIXME (missing): Rename to clone-ecj.
 stamps/extract-ecj.stamp: $(OPENJDK_TREE)
 	if ! test -d openjdk-ecj ; then \
-	    cp -pPRl openjdk openjdk-ecj ; \
+	    cp -pPR openjdk openjdk-ecj ; \
 	fi
 	mkdir -p stamps
 	touch stamps/extract-ecj.stamp
@@ -1642,11 +1642,15 @@ clean-patch-ecj:
 # ===========================
 
 stamps/native-ecj.stamp: 
-	mkdir -p stamps ; \
-	if test "x$(GCJ)" != "xno"; then \
-	  $(GCJ) $(IT_CFLAGS) -Wl,-Bsymbolic -findirect-dispatch -o native-ecj \
-	    --main=org.eclipse.jdt.internal.compiler.batch.Main ${ECJ_JAR} ; \
-	fi ; \
+	mkdir -p stamps ; \
+	if test -n "${ADK_ECJ}" && test -x "${ADK_ECJ}"; then \
+		ln -s "${ADK_ECJ}" native-ecj; \
+	else \
+		if test "x$(GCJ)" != "xno"; then \
+		  $(GCJ) $(IT_CFLAGS) -Wl,-Bsymbolic -findirect-dispatch -o native-ecj \
+		    --main=org.eclipse.jdt.internal.compiler.batch.Main ${ECJ_JAR} ; \
+		fi ; \
+	fi; \
 	touch stamps/native-ecj.stamp
 
 clean-native-ecj:
