--- w-openjdk-6.orig/openjdk-6/hotspot/src/share/vm/shark/llvmHeaders.hpp	2011-02-28 17:03:21.000000000 +0100
+++ w-openjdk-6/openjdk-6/hotspot/src/share/vm/shark/llvmHeaders.hpp	2011-04-29 18:07:20.171550601 +0200
@@ -43,7 +43,7 @@
 #include <llvm/ModuleProvider.h>
 #endif
 #include <llvm/Support/IRBuilder.h>
-#include <llvm/System/Threading.h>
+#include <llvm/Support/Threading.h>
 #include <llvm/Target/TargetSelect.h>
 #include <llvm/Type.h>
 #include <llvm/ExecutionEngine/JITMemoryManager.h>
@@ -52,7 +52,7 @@
 #include <llvm/ExecutionEngine/JIT.h>
 #include <llvm/ADT/StringMap.h>
 #include <llvm/Support/Debug.h>
-#include <llvm/System/Host.h>
+#include <llvm/Support/Host.h>
 #endif
 
 #include <map>
