 
 disable -Werror

--- w-openjdk-6.orig/openjdk-6/hotspot/make/linux/makefiles/gcc.make	2011-02-28 17:03:13.000000000 +0100
+++ w-openjdk-6/openjdk-6/hotspot/make/linux/makefiles/gcc.make	2011-08-19 18:18:55.997423368 +0200
@@ -108,9 +108,6 @@
   CFLAGS += -pipe
 endif
 
-# Compiler warnings are treated as errors
-WARNINGS_ARE_ERRORS = -Werror
-
 # Except for a few acceptable ones
 # Since GCC 4.3, -Wconversion has changed its meanings to warn these implicit
 # conversions which might affect the values. To avoid that, we need to turn
