diff -Nur icedtea-2.3.9.orig/configure icedtea-2.3.9/configure
--- icedtea-2.3.9.orig/configure	2013-04-21 22:48:30.000000000 +0200
+++ icedtea-2.3.9/configure	2013-05-13 12:43:02.000000000 +0200
@@ -7159,6 +7159,11 @@
       INSTALL_ARCH_DIR=mipsel
       JRE_ARCH_DIR=mipsel
        ;;
+    mips64el)
+      BUILD_ARCH_DIR=mipsel
+      INSTALL_ARCH_DIR=mipsel
+      JRE_ARCH_DIR=mipsel
+       ;;
     powerpc)
       BUILD_ARCH_DIR=ppc
       INSTALL_ARCH_DIR=ppc
