diff -Nur binutils-2.26.1.orig/bfd/elf32-cris.c binutils-2.26.1/bfd/elf32-cris.c
--- binutils-2.26.1.orig/bfd/elf32-cris.c	2015-11-13 09:27:40.000000000 +0100
+++ binutils-2.26.1/bfd/elf32-cris.c	2016-07-09 19:19:49.305293884 +0200
@@ -2728,8 +2728,8 @@
 
   /* A GOTPLT reloc, when activated, is supposed to be included into
      the PLT refcount.  */
-  BFD_ASSERT (h->gotplt_refcount == 0
-	      || h->gotplt_refcount <= h->root.plt.refcount);
+  /* BFD_ASSERT (h->gotplt_refcount == 0
+	      || h->gotplt_refcount <= h->root.plt.refcount); */
 
   /* If nobody wanted a GOTPLT with this symbol, we're done.  */
   if (h->gotplt_refcount <= 0)
