From:
https://sourceware.org/bugzilla/show_bug.cgi?id=21017
diff -Nur binutils-2.26.1.orig/bfd/elf32-microblaze.c binutils-2.26.1/bfd/elf32-microblaze.c
--- binutils-2.26.1.orig/bfd/elf32-microblaze.c	2015-11-13 09:27:40.000000000 +0100
+++ binutils-2.26.1/bfd/elf32-microblaze.c	2017-01-03 06:31:27.305957954 +0100
@@ -2402,6 +2402,7 @@
           tls_type |= (TLS_TLS | TLS_LD);
         dogottls:
           sec->has_tls_reloc = 1;
+	case R_MICROBLAZE_GOTOFF_64:
         case R_MICROBLAZE_GOT_64:
           if (htab->sgot == NULL)
             {
