diff -Nur linux-4.14.44.orig/arch/x86/entry/vdso/vdso2c.c linux-4.14.44/arch/x86/entry/vdso/vdso2c.c
--- linux-4.14.44.orig/arch/x86/entry/vdso/vdso2c.c	2018-05-25 16:18:02.000000000 +0200
+++ linux-4.14.44/arch/x86/entry/vdso/vdso2c.c	2018-05-31 02:09:00.000000000 +0200
@@ -64,7 +64,15 @@
 #include <tools/le_byteshift.h>
 
 #include <linux/elf.h>
+#ifdef __linux__
 #include <linux/types.h>
+#else
+typedef _Bool			bool;
+enum {
+        false   = 0,
+        true    = 1
+};
+#endif
 
 const char *outfilename;
 
@@ -86,6 +94,7 @@
 	sym_hvclock_page,
 };
 
+
 struct vdso_sym {
 	const char *name;
 	bool export;
