diff -Nur linux-3.11.5.orig/scripts/Makefile.headersinst linux-3.11.5/scripts/Makefile.headersinst
--- linux-3.11.5.orig/scripts/Makefile.headersinst	2013-10-14 03:14:45.000000000 +0200
+++ linux-3.11.5/scripts/Makefile.headersinst	2013-10-15 16:33:10.000000000 +0200
@@ -107,7 +107,6 @@
 
 targets += $(install-file)
 $(install-file): scripts/headers_install.sh $(input-files1) $(input-files2) $(input-files3) FORCE
-	$(if $(unwanted),$(call cmd,remove),)
 	$(if $(wildcard $(dir $@)),,$(shell mkdir -p $(dir $@)))
 	$(call if_changed,install)
 
