diff -Nur linux-3.11.5.orig/arch/mips/include/uapi/asm/sgidefs.h linux-3.11.5/arch/mips/include/uapi/asm/sgidefs.h
--- linux-3.11.5.orig/arch/mips/include/uapi/asm/sgidefs.h	2013-10-14 03:14:45.000000000 +0200
+++ linux-3.11.5/arch/mips/include/uapi/asm/sgidefs.h	2013-11-08 22:01:28.000000000 +0100
@@ -11,14 +11,6 @@
 #define __ASM_SGIDEFS_H
 
 /*
- * Using a Linux compiler for building Linux seems logic but not to
- * everybody.
- */
-#ifndef __linux__
-#error Use a Linux compiler or give up.
-#endif
-
-/*
  * Definitions for the ISA levels
  *
  * With the introduction of MIPS32 / MIPS64 instruction sets definitions
