diff -Nur linux-3.12.6.orig/arch/arm/Kconfig linux-3.12.6/arch/arm/Kconfig
--- linux-3.12.6.orig/arch/arm/Kconfig	2013-12-20 16:51:33.000000000 +0100
+++ linux-3.12.6/arch/arm/Kconfig	2013-12-28 19:29:33.000000000 +0100
@@ -6,7 +6,6 @@
 	select ARCH_HAS_TICK_BROADCAST if GENERIC_CLOCKEVENTS_BROADCAST
 	select ARCH_HAVE_CUSTOM_GPIO_H
 	select ARCH_WANT_IPC_PARSE_VERSION
-	select BUILDTIME_EXTABLE_SORT if MMU
 	select CLONE_BACKWARDS
 	select CPU_PM if (SUSPEND || CPU_IDLE)
 	select DCACHE_WORD_ACCESS if (CPU_V6 || CPU_V6K || CPU_V7) && !CPU_BIG_ENDIAN && MMU
diff -Nur linux-3.12.6.orig/arch/mips/Kconfig linux-3.12.6/arch/mips/Kconfig
--- linux-3.12.6.orig/arch/mips/Kconfig	2013-12-20 16:51:33.000000000 +0100
+++ linux-3.12.6/arch/mips/Kconfig	2013-12-28 19:30:06.000000000 +0100
@@ -35,7 +35,6 @@
 	select HAVE_MEMBLOCK_NODE_MAP
 	select ARCH_DISCARD_MEMBLOCK
 	select GENERIC_SMP_IDLE_THREAD
-	select BUILDTIME_EXTABLE_SORT
 	select GENERIC_CLOCKEVENTS
 	select GENERIC_CMOS_UPDATE
 	select HAVE_MOD_ARCH_SPECIFIC
diff -Nur linux-3.12.6.orig/arch/x86/Kconfig linux-3.12.6/arch/x86/Kconfig
--- linux-3.12.6.orig/arch/x86/Kconfig	2013-12-20 16:51:33.000000000 +0100
+++ linux-3.12.6/arch/x86/Kconfig	2013-12-28 19:29:50.000000000 +0100
@@ -100,7 +100,6 @@
 	select GENERIC_SMP_IDLE_THREAD
 	select ARCH_WANT_IPC_PARSE_VERSION if X86_32
 	select HAVE_ARCH_SECCOMP_FILTER
-	select BUILDTIME_EXTABLE_SORT
 	select GENERIC_CMOS_UPDATE
 	select HAVE_ARCH_SOFT_DIRTY
 	select CLOCKSOURCE_WATCHDOG
