diff -Nur uClibc-ng-1.0.21.orig/Makefile.in uClibc-ng-1.0.21/Makefile.in
--- uClibc-ng-1.0.21.orig/Makefile.in	2016-12-26 09:12:14.000000000 +0100
+++ uClibc-ng-1.0.21/Makefile.in	2016-12-27 18:33:22.312597987 +0100
@@ -282,7 +282,7 @@
 HEADERS_RM-$(UCLIBC_HAS_BACKTRACE)           += execinfo.h
 HEADERS_RM-$(UCLIBC_HAS_LIBICONV)            += iconv.h
 HEADERS_RM-$(UCLIBC_HAS_LIBINTL)             += intl.h
-HEADERS_RM-$(UCLIBC_HAS_LOCALE)              += iconv.h bits/uClibc_ctype.h
+HEADERS_RM-$(UCLIBC_HAS_LOCALE)              += bits/uClibc_ctype.h
 HEADERS_RM-$(UCLIBC_HAS_PTY)                 += pty.h
 HEADERS_RM-$(UCLIBC_HAS_REALTIME)            += mqueue.h bits/mqueue.h sched.h \
 	bits/sched.h \
