diff -Nur linux-3.10.33.orig/init/initramfs.c linux-3.10.33/init/initramfs.c
--- linux-3.10.33.orig/init/initramfs.c	2014-03-07 06:58:45.000000000 +0100
+++ linux-3.10.33/init/initramfs.c	2014-03-15 18:32:07.240436750 +0100
@@ -622,6 +622,9 @@
 		 */
 		load_default_modules();
 	}
+#ifdef CONFIG_DEVTMPFS_MOUNT
+	devtmpfs_mount("dev");
+#endif
 	return 0;
 }
 rootfs_initcall(populate_rootfs);
diff -Nur linux-3.10.33.orig/init/main.c linux-3.10.33/init/main.c
--- linux-3.10.33.orig/init/main.c	2014-03-07 06:58:45.000000000 +0100
+++ linux-3.10.33/init/main.c	2014-03-15 18:32:07.240436750 +0100
@@ -885,6 +885,8 @@
 	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
 		pr_err("Warning: unable to open an initial console.\n");
 
+	printk(KERN_WARNING "Starting Linux (built with OpenADK).\n");
+
 	(void) sys_dup(0);
 	(void) sys_dup(0);
 	/*
@@ -893,7 +895,7 @@
 	 */
 
 	if (!ramdisk_execute_command)
-		ramdisk_execute_command = "/init";
+		ramdisk_execute_command = "/sbin/init";
 
 	if (sys_access((const char __user *) ramdisk_execute_command, 0) != 0) {
 		ramdisk_execute_command = NULL;
