diff -Nur linux-2.6.31.4.orig/arch/cris/mm/init.c linux-2.6.31.4/arch/cris/mm/init.c
--- linux-2.6.31.4.orig/arch/cris/mm/init.c	2009-10-12 22:15:40.000000000 +0200
+++ linux-2.6.31.4/arch/cris/mm/init.c	2009-10-25 12:59:24.418546156 +0100
@@ -80,3 +80,10 @@
         printk (KERN_INFO "Freeing unused kernel memory: %luk freed\n",
 		(unsigned long)((&__init_end - &__init_begin) >> 10));
 }
+
+#ifdef CONFIG_BLK_DEV_INITRD
+void free_initrd_mem(unsigned long start, unsigned long end)
+{
+	return 0;
+}
+#endif
