diff -Nur linux-2.6.36.orig/drivers/gpu/drm/Kconfig linux-2.6.36/drivers/gpu/drm/Kconfig
--- linux-2.6.36.orig/drivers/gpu/drm/Kconfig	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36/drivers/gpu/drm/Kconfig	2010-11-24 23:08:39.707247343 +0100
@@ -79,21 +79,6 @@
 	  selected, the module will be called i810.  AGP support is required
 	  for this driver to work.
 
-choice
-	prompt "Intel 830M, 845G, 852GM, 855GM, 865G"
-	depends on DRM && AGP && AGP_INTEL
-	optional
-
-config DRM_I830
-	tristate "i830 driver"
-	help
-	  Choose this option if you have a system that has Intel 830M, 845G,
-	  852GM, 855GM or 865G integrated graphics.  If M is selected, the
-	  module will be called i830.  AGP support is required for this driver
-	  to work. This driver is used by the older X releases X.org 6.7 and
-	  XFree86 4.3. If unsure, build this and i915 as modules and the X server
-	  will load the correct one.
-
 config DRM_I915
 	tristate "i915 driver"
 	depends on AGP_INTEL
@@ -127,8 +112,6 @@
 	  the driver to bind to PCI devices, which precludes loading things
 	  like intelfb.
 
-endchoice
-
 config DRM_MGA
 	tristate "Matrox g200/g400"
 	depends on DRM && PCI
