--- digitemp-3.3.2.orig/Makefile	2004-02-04 15:17:10.000000000 +0100
+++ digitemp-3.3.2/Makefile	2006-08-16 20:17:27.000000000 +0200
@@ -15,8 +15,6 @@
 
 VERSION = 3.3.2
 
-
-CC	= gcc
 CFLAGS	= -I./src -I./userial -O2 # -g -Wall
 LIBS	= -lm
 
@@ -57,7 +55,7 @@
   CFLAGS += -DLINUX
 
   # Set LOCK to yes for serial port locking support
-  LOCK = yes
+  #LOCK = yes
 
 endif
 
@@ -123,7 +121,7 @@
 		$(CC) $(OBJS) $(ONEWIREOBJS) $(DS9097UOBJS) -o digitemp_DS9097U $(LIBS)
 
 ds2490:		$(OBJS) $(HDRS) $(ONEWIREOBJS) $(ONEWIREHDRS) $(DS2490OBJS)
-		$(CC) $(OBJS) $(ONEWIREOBJS) $(DS2490OBJS) -o digitemp_DS2490 $(LIBS)
+		$(CC) $(LDFLAGS) $(OBJS) $(ONEWIREOBJS) $(DS2490OBJS) -o digitemp_DS2490 $(LIBS)
 
 
 # Clean up the object files and the sub-directory for distributions
