diff -Nur linux-3.14.3.orig/arch/mips/pci/pci.c linux-3.14.3/arch/mips/pci/pci.c
--- linux-3.14.3.orig/arch/mips/pci/pci.c	2014-05-06 16:59:58.000000000 +0200
+++ linux-3.14.3/arch/mips/pci/pci.c	2014-05-11 10:13:26.000000000 +0200
@@ -166,20 +166,10 @@
 
 void register_pci_controller(struct pci_controller *hose)
 {
-	struct resource *parent;
-
-	parent = hose->mem_resource->parent;
-	if (!parent)
-		parent = &iomem_resource;
-
-	if (request_resource(parent, hose->mem_resource) < 0)
+	if (request_resource(&iomem_resource, hose->mem_resource) < 0)
 		goto out;
 
-	parent = hose->io_resource->parent;
-	if (!parent)
-		parent = &ioport_resource;
-
-	if (request_resource(parent, hose->io_resource) < 0) {
+	if (request_resource(&ioport_resource, hose->io_resource) < 0) {
 		release_resource(hose->mem_resource);
 		goto out;
 	}
