diff -Nur eglibc-2.16.orig/libc/Makeconfig eglibc-2.16/libc/Makeconfig
--- eglibc-2.16.orig/libc/Makeconfig	2012-07-31 09:00:21.000000000 +0200
+++ eglibc-2.16/libc/Makeconfig	2012-08-30 13:01:34.000000000 +0200
@@ -521,9 +521,9 @@
 else
   libunwind = -lunwind
 endif
-libgcc_eh := -Wl,--as-needed -lgcc_s $(libunwind) -Wl,--no-as-needed
+libgcc_eh := -Wl,--as-needed $(libunwind) -Wl,--no-as-needed
 gnulib := -lgcc $(libgcc_eh)
-static-gnulib := -lgcc -lgcc_eh $(libunwind)
+static-gnulib := -lgcc $(libunwind)
 libc.so-gnulib := -lgcc
 endif
 +preinit = $(addprefix $(csu-objpfx),crti.o)
