diff -Nur eglibc-2.12.1.orig/libc/sysdeps/sparc/sparc64/Makefile eglibc-2.12.1/libc/sysdeps/sparc/sparc64/Makefile
--- eglibc-2.12.1.orig/libc/sysdeps/sparc/sparc64/Makefile	2010-09-28 19:13:11.000000000 +0200
+++ eglibc-2.12.1/libc/sysdeps/sparc/sparc64/Makefile	2011-01-02 12:43:41.009426907 +0100
@@ -1,3 +1,5 @@
+sysdep-CFLAGS += -mcpu=v9 -Wa,-Av9a
+
 ifeq ($(subdir),csu)
 sysdep_routines += hp-timing
 elide-routines.os += hp-timing
@@ -10,3 +12,11 @@
 ifeq ($(subdir),string)
 sysdep_routines += align-cpy
 endif
+
+ASFLAGS-.o += -Wa,-Av9a
+ASFLAGS-.os += -Wa,-Av9a
+ASFLAGS-.op += -Wa,-Av9a
+ASFLAGS-.og += -Wa,-Av9a
+ASFLAGS-.ob += -Wa,-Av9a
+ASFLAGS-.oS += -Wa,-Av9a
+
