diff -Nur gcc-6.1.0.orig/libgcc/Makefile.in gcc-6.1.0/libgcc/Makefile.in
--- gcc-6.1.0.orig/libgcc/Makefile.in	2016-02-25 13:23:52.000000000 +0100
+++ gcc-6.1.0/libgcc/Makefile.in	2016-08-11 21:15:13.841764686 +0200
@@ -932,7 +932,11 @@
 	{ $(NM) $(SHLIB_NM_FLAGS) $(libgcc-s-objects); echo %%; \
 	  cat libgcc.map.in; \
 	} | $(AWK) -f $(SHLIB_MKMAP) $(SHLIB_MKMAP_OPTS) > tmp-$@
+ifeq ($(cpu_type),bfin)
+	sed -e 's/__/_/' tmp-$@ > $@
+else
 	mv tmp-$@ $@
+endif
 libgcc_s$(SHLIB_EXT): libgcc.map
 mapfile = libgcc.map
 endif
