diff -Nur gcc-6.2.0.orig/gcc/config/s390/linux.h gcc-6.2.0/gcc/config/s390/linux.h
--- gcc-6.2.0.orig/gcc/config/s390/linux.h	2016-01-04 15:30:50.000000000 +0100
+++ gcc-6.2.0/gcc/config/s390/linux.h	2016-11-23 00:05:47.186562086 +0100
@@ -63,6 +63,11 @@
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
 #define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
 
+#undef MUSL_DYNAMIC_LINKER32
+#define MUSL_DYNAMIC_LINKER32 "/lib/ld-musl-s390.so.1"
+#undef MUSL_DYNAMIC_LINKER64
+#define MUSL_DYNAMIC_LINKER64 "/lib/ld-musl-s390x.so.1"
+
 #undef  LINK_SPEC
 #define LINK_SPEC \
   "%{m31:-m elf_s390}%{m64:-m elf64_s390} \
