diff -ruN socat-1.4-old/Makefile.in socat-1.4-new/Makefile.in
--- socat-1.7.2.2.orig/Makefile.in	2011-12-06 08:45:03.000000000 +0100
+++ socat-1.7.2.2/Makefile.in	2013-12-24 16:37:21.000000000 +0100
@@ -38,7 +38,8 @@ INSTALL = @INSTALL@
 
 #0 CFLAGS = @CFLAGS@ $(CCOPTS) $(DEFS) $(INCLS)
 CFLAGS = @CFLAGS@ $(CCOPTS) $(DEFS) $(CPPFLAGS)
-CLIBS = $(LIBS)
+LDFLAGS = @LDFLAGS@
+CLIBS = $(LDFLAGS) $(LIBS)
 #CLIBS = $(LIBS) -lm -lefence
 XIOSRCS = xioinitialize.c xiohelp.c xioparam.c xiodiag.c xioopen.c xioopts.c \
 	xiosignal.c xiosigchld.c xioread.c xiowrite.c \
