diff -pur pcc-libs-1.0.0~/csu/linux/Makefile pcc-libs-1.0.0/csu/linux/Makefile
--- pcc-libs-1.0.0.orig/csu/linux/Makefile	2009-08-13 11:16:46.000000000 +0200
+++ pcc-libs-1.0.0/csu/linux/Makefile	2011-04-07 20:36:10.068600370 +0200
@@ -4,7 +4,7 @@ VERSION ?= 0.9.9
 PCCLIBDIR ?= /usr/local/lib/pcc/$(TARGET)/$(VERSION)/lib
 SRCDIR ?= .
 
-CFLAGS = -I$(SRCDIR)/$(MACH) -I$(SRCDIR) -Wall -Wstrict-prototypes -Wmissing-prototypes
+CFLAGS += -I$(SRCDIR)/$(MACH) -I$(SRCDIR) -Wall -Wstrict-prototypes -Wmissing-prototypes
 
 OBJS = crt0.o crt1.o gcrt1.o crti.o crtn.o crtbegin.o crtend.o
 
