diff -Nur linux-3.14.3.orig/arch/mips/pci/pci-rc32434.c linux-3.14.3/arch/mips/pci/pci-rc32434.c
--- linux-3.14.3.orig/arch/mips/pci/pci-rc32434.c	2014-05-06 16:59:58.000000000 +0200
+++ linux-3.14.3/arch/mips/pci/pci-rc32434.c	2014-05-13 17:15:48.000000000 +0200
@@ -53,7 +53,6 @@
 	.start = 0x50000000,
 	.end = 0x5FFFFFFF,
 	.flags = IORESOURCE_MEM,
-	.parent = &rc32434_res_pci_mem1,
 	.sibling = NULL,
 	.child = &rc32434_res_pci_mem2
 };
