diff -Nur binutils-2.28.orig/bfd/elf32-cris.c binutils-2.28/bfd/elf32-cris.c
--- binutils-2.28.orig/bfd/elf32-cris.c	2017-03-02 09:23:53.000000000 +0100
+++ binutils-2.28/bfd/elf32-cris.c	2017-03-05 09:38:18.895667626 +0100
@@ -2713,11 +2713,6 @@
 {
   struct bfd_link_info *info = (struct bfd_link_info *) p;
 
-  /* A GOTPLT reloc, when activated, is supposed to be included into
-     the PLT refcount.  */
-  BFD_ASSERT (h->gotplt_refcount == 0
-	      || h->gotplt_refcount <= h->root.plt.refcount);
-
   /* If nobody wanted a GOTPLT with this symbol, we're done.  */
   if (h->gotplt_refcount <= 0)
     return TRUE;
