diff -Nur linux-3.15.1.orig/arch/mips/rb532/devices.c linux-3.15.1/arch/mips/rb532/devices.c
--- linux-3.15.1.orig/arch/mips/rb532/devices.c	2014-06-16 22:44:27.000000000 +0200
+++ linux-3.15.1/arch/mips/rb532/devices.c	2014-06-28 10:27:20.418061077 +0200
@@ -223,6 +223,7 @@
 
 static struct plat_serial8250_port rb532_uart_res[] = {
 	{
+		.type           = PORT_16550A,
 		.membase	= (char *)KSEG1ADDR(REGBASE + UART0BASE),
 		.irq		= UART0_IRQ,
 		.regshift	= 2,
