diff -Nur linux-3.13.3.orig/drivers/net/ethernet/xilinx/xilinx_axienet_main.c linux-3.13.3/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
--- linux-3.13.3.orig/drivers/net/ethernet/xilinx/xilinx_axienet_main.c	2014-02-13 23:00:14.000000000 +0100
+++ linux-3.13.3/drivers/net/ethernet/xilinx/xilinx_axienet_main.c	2014-02-24 08:03:57.000000000 +0100
@@ -25,6 +25,7 @@
 #include <linux/init.h>
 #include <linux/module.h>
 #include <linux/netdevice.h>
+#include <linux/of_irq.h>
 #include <linux/of_mdio.h>
 #include <linux/of_platform.h>
 #include <linux/of_address.h>
