diff -Nur linux-2.6.32.orig/include/scsi/Kbuild linux-2.6.32/include/scsi/Kbuild
--- linux-2.6.32.orig/include/scsi/Kbuild	2009-12-03 04:51:21.000000000 +0100
+++ linux-2.6.32/include/scsi/Kbuild	2009-12-03 19:46:03.000000000 +0100
@@ -1,3 +1,5 @@
+header-y += sg.h
+header-y += scsi_ioctl.h
 header-y += scsi.h
 header-y += scsi_netlink.h
 header-y += scsi_netlink_fc.h
diff -Nur linux-2.6.32.orig/include/scsi/scsi.h linux-2.6.32/include/scsi/scsi.h
--- linux-2.6.32.orig/include/scsi/scsi.h	2009-12-03 04:51:21.000000000 +0100
+++ linux-2.6.32/include/scsi/scsi.h	2009-12-03 19:47:49.000000000 +0100
@@ -143,6 +143,8 @@
 
 #define SCSI_MAX_VARLEN_CDB_SIZE 260
 
+#ifdef __KERNEL__
+
 /* defined in T10 SCSI Primary Commands-2 (SPC2) */
 struct scsi_varlen_cdb_hdr {
 	u8 opcode;        /* opcode always == VARIABLE_LENGTH_CMD */
@@ -169,6 +171,8 @@
 		scsi_varlen_cdb_length(cmnd) : COMMAND_SIZE(cmnd[0]);
 }
 
+#endif
+
 /*
  *  SCSI Architecture Model (SAM) Status codes. Taken from SAM-3 draft
  *  T10/1561-D Revision 4 Draft dated 7th November 2002.
@@ -284,8 +288,10 @@
 	SCSI_PROTOCOL_UNSPEC = 0xf, /* No specific protocol */
 };
 
+#ifdef __KERNEL__
 /* Returns a human-readable name for the device */
 extern const char * scsi_device_type(unsigned type);
+#endif
 
 /*
  * standard mode-select header prepended to all mode-select commands
