diff -Nur linux-3.11.10.orig/drivers/net/ethernet/xilinx/xilinx_emaclite.c linux-3.11.10/drivers/net/ethernet/xilinx/xilinx_emaclite.c
--- linux-3.11.10.orig/drivers/net/ethernet/xilinx/xilinx_emaclite.c	2013-11-29 19:42:37.000000000 +0100
+++ linux-3.11.10/drivers/net/ethernet/xilinx/xilinx_emaclite.c	2013-12-23 20:01:14.000000000 +0100
@@ -1282,6 +1282,7 @@
 	{ .compatible = "xlnx,opb-ethernetlite-1.01.b", },
 	{ .compatible = "xlnx,xps-ethernetlite-1.00.a", },
 	{ .compatible = "xlnx,xps-ethernetlite-2.00.a", },
+	{ .compatible = "xlnx,xps-ethernetlite-2.00.b", },
 	{ .compatible = "xlnx,xps-ethernetlite-2.01.a", },
 	{ .compatible = "xlnx,xps-ethernetlite-3.00.a", },
 	{ /* end of list */ },
