diff -Nur linux-3.14.4.orig/drivers/base/regmap/Kconfig linux-3.14.4/drivers/base/regmap/Kconfig
--- linux-3.14.4.orig/drivers/base/regmap/Kconfig	2014-05-13 13:33:14.000000000 +0200
+++ linux-3.14.4/drivers/base/regmap/Kconfig	2014-05-21 13:46:48.928985115 +0200
@@ -10,16 +10,16 @@
 	bool
 
 config REGMAP_I2C
-	tristate
+	boolean
 
 config REGMAP_SPI
-	tristate
+	boolean
 
 config REGMAP_SPMI
-	tristate
+	boolean
 
 config REGMAP_MMIO
-	tristate
+	boolean
 
 config REGMAP_IRQ
 	bool
