diff -Nur linux-3.5.orig/arch/mips/Kconfig linux-3.5/arch/mips/Kconfig
--- linux-3.5.orig/arch/mips/Kconfig	2012-07-21 22:58:29.000000000 +0200
+++ linux-3.5/arch/mips/Kconfig	2012-07-30 12:00:26.000000000 +0200
@@ -31,7 +31,6 @@
 	select HAVE_MEMBLOCK_NODE_MAP
 	select ARCH_DISCARD_MEMBLOCK
 	select GENERIC_SMP_IDLE_THREAD
-	select BUILDTIME_EXTABLE_SORT
 	select GENERIC_CLOCKEVENTS
 	select GENERIC_CMOS_UPDATE
 
