diff -Nur linux-3.11.5.orig/arch/arm/Kconfig linux-3.11.5/arch/arm/Kconfig
--- linux-3.11.5.orig/arch/arm/Kconfig	2013-10-14 03:14:45.000000000 +0200
+++ linux-3.11.5/arch/arm/Kconfig	2013-10-16 10:36:41.000000000 +0200
@@ -6,7 +6,6 @@
 	select ARCH_HAVE_CUSTOM_GPIO_H
 	select ARCH_HAS_TICK_BROADCAST if GENERIC_CLOCKEVENTS_BROADCAST
 	select ARCH_WANT_IPC_PARSE_VERSION
-	select BUILDTIME_EXTABLE_SORT if MMU
 	select CPU_PM if (SUSPEND || CPU_IDLE)
 	select DCACHE_WORD_ACCESS if (CPU_V6 || CPU_V6K || CPU_V7) && !CPU_BIG_ENDIAN && MMU
 	select GENERIC_ATOMIC64 if (CPU_V7M || CPU_V6 || !CPU_32v6K || !AEABI)
diff -Nur linux-3.11.5.orig/arch/mips/Kconfig linux-3.11.5/arch/mips/Kconfig
--- linux-3.11.5.orig/arch/mips/Kconfig	2013-10-14 03:14:45.000000000 +0200
+++ linux-3.11.5/arch/mips/Kconfig	2013-10-16 10:36:41.000000000 +0200
@@ -36,7 +36,6 @@
 	select HAVE_MEMBLOCK_NODE_MAP
 	select ARCH_DISCARD_MEMBLOCK
 	select GENERIC_SMP_IDLE_THREAD
-	select BUILDTIME_EXTABLE_SORT
 	select GENERIC_CLOCKEVENTS
 	select GENERIC_CMOS_UPDATE
 	select HAVE_MOD_ARCH_SPECIFIC
diff -Nur linux-3.11.5.orig/arch/x86/Kconfig linux-3.11.5/arch/x86/Kconfig
--- linux-3.11.5.orig/arch/x86/Kconfig	2013-10-14 03:14:45.000000000 +0200
+++ linux-3.11.5/arch/x86/Kconfig	2013-10-16 10:36:55.000000000 +0200
@@ -101,7 +101,6 @@
 	select GENERIC_SMP_IDLE_THREAD
 	select ARCH_WANT_IPC_PARSE_VERSION if X86_32
 	select HAVE_ARCH_SECCOMP_FILTER
-	select BUILDTIME_EXTABLE_SORT
 	select GENERIC_CMOS_UPDATE
 	select HAVE_ARCH_SOFT_DIRTY
 	select CLOCKSOURCE_WATCHDOG
