diff -Nur linux-4.4.88.orig/arch/h8300/Kconfig.cpu linux-4.4.88/arch/h8300/Kconfig.cpu
--- linux-4.4.88.orig/arch/h8300/Kconfig.cpu	2017-09-13 23:10:05.000000000 +0200
+++ linux-4.4.88/arch/h8300/Kconfig.cpu	2017-09-17 21:15:32.584654434 +0200
@@ -96,4 +96,8 @@
         hex "Load offset"
 	default 0
 
+config RAMBASE
+	hex "RAM base address"
+	default 0x400000
+
 endmenu
diff -Nur linux-4.4.96.orig/arch/h8300/boot/dts/h8300h_sim.dts linux-4.4.96/arch/h8300/boot/dts/h8300h_sim.dts
--- linux-4.4.96.orig/arch/h8300/boot/dts/h8300h_sim.dts	2017-11-02 09:40:50.000000000 +0100
+++ linux-4.4.96/arch/h8300/boot/dts/h8300h_sim.dts	2017-11-18 21:59:54.665879649 +0100
@@ -37,7 +37,7 @@
 
 	memory@400000 {
 		device_type = "memory";
-		reg = <0x400000 0x400000>;
+		reg = <0x400000 0x800000>;
 	};
 
 	cpus {
diff -Nur linux-4.4.96.orig/drivers/tty/serial/sh-sci.c linux-4.4.96/drivers/tty/serial/sh-sci.c
--- linux-4.4.96.orig/drivers/tty/serial/sh-sci.c	2017-11-02 09:40:50.000000000 +0100
+++ linux-4.4.96/drivers/tty/serial/sh-sci.c	2017-11-18 22:15:05.951976319 +0100
@@ -898,8 +898,6 @@
 		/* overrun error */
 		if (tty_insert_flip_char(tport, 0, TTY_OVERRUN))
 			copied++;
-
-		dev_notice(port->dev, "overrun error\n");
 	}
 
 	if (status & SCxSR_FER(port)) {
@@ -972,7 +970,6 @@
 		tty_insert_flip_char(tport, 0, TTY_OVERRUN);
 		tty_flip_buffer_push(tport);
 
-		dev_dbg(port->dev, "overrun error\n");
 		copied++;
 	}
 
