--- libdnet-1.10.orig/dnet-config.in	2001-10-19 03:29:00.000000000 +0200
+++ libdnet-1.10/dnet-config.in	2011-01-14 11:44:00.000000000 +0100
@@ -45,10 +45,18 @@ while test $# -gt 0; do
 done
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      includes=
+      if test "@includedir@" != "/usr/include" ; then
+             includes=-I@includedir@
+      fi
+      echo $includes
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldnet @LIBS@
+      libs=
+      if test "@libdir@" != "/usr/lib" ; then
+             libs=-I@libdir@
+      fi
+      echo $libs -ldnet @LIBS@
 fi      
 
